{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import numpy as np\n",
    "from pandas import Series"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Open filtered fie from control data from diseasea data\n",
    "df = pd.read_csv('Asthma_final_variant.txt', sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To samples contains this mutation\n",
    "df['Sample_Count'] = (df.iloc[:,9:] != './.:.:.:.:.').sum(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    2\n",
       "1    1\n",
       "2    1\n",
       "3    4\n",
       "4    1\n",
       "Name: Sample_Count, dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Sample_Count'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove columns starting with SRR from main variable\n",
    "df = df[df.columns.drop(list(df.filter(regex='SRR')))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>45.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BaseQRankSum=-1.15;ExcessHet=3.0103;FS=0;MQ=60...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14673</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>71.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=2.303...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15015</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>259.65</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=34.61;SOR=2.833...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17385</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>57.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BaseQRankSum=-0.674;ExcessHet=3.0103;FS=0;MQ=6...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM    POS ID REF ALT    QUAL FILTER  \\\n",
       "0  chr1  14522  .   G   A  118.64   PASS   \n",
       "1  chr1  14590  .   G   A   45.64   PASS   \n",
       "2  chr1  14673  .   G   C   71.32   PASS   \n",
       "3  chr1  15015  .   G   C  259.65   PASS   \n",
       "4  chr1  17385  .   G   A   57.64   PASS   \n",
       "\n",
       "                                                INFO          FORMAT  \\\n",
       "0  ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "1  BaseQRankSum=-1.15;ExcessHet=3.0103;FS=0;MQ=60...  GT:AD:DP:GQ:PL   \n",
       "2  ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "3  ExcessHet=3.0103;FS=0;MQ=60;QD=34.61;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "4  BaseQRankSum=-0.674;ExcessHet=3.0103;FS=0;MQ=6...  GT:AD:DP:GQ:PL   \n",
       "\n",
       "   Sample_Count  \n",
       "0             2  \n",
       "1             1  \n",
       "2             1  \n",
       "3             4  \n",
       "4             1  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1125712"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df.axes[0]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# info column have many consequences seperated by comma for one mutation to split (comma) that add to next next row by keeping all data\n",
    "s = df['INFO'].str.split(',').apply(Series, 1).stack() # split by comma\n",
    "s.index = s.index.droplevel(-1) # index splitted column\n",
    "s.name = 'INFO' # rename column as INFO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6869861"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(s.axes[0]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...\n",
       "0    A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...\n",
       "0    A|intron_variant&non_coding_transcript_variant...\n",
       "0    A|downstream_gene_variant|MODIFIER|MIR6859-1|E...\n",
       "0    A|regulatory_region_variant|MODIFIER|||Regulat...\n",
       "Name: INFO, dtype: object"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To delete orignal INFO column from main df file\n",
    "del df['INFO']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# join spliited info column with main df dataframe\n",
    "df1 = df.join(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample_Count</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM    POS ID REF ALT    QUAL FILTER          FORMAT  Sample_Count  \\\n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "\n",
       "                                                INFO  \n",
       "0  ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...  \n",
       "0  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "0  A|intron_variant&non_coding_transcript_variant...  \n",
       "0  A|downstream_gene_variant|MODIFIER|MIR6859-1|E...  \n",
       "0  A|regulatory_region_variant|MODIFIER|||Regulat...  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Again split INFo column by bar(|) to seperate all the colums\n",
    "k = df1.INFO.str.split(\"|\",expand=True,)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>54</th>\n",
       "      <th>55</th>\n",
       "      <th>56</th>\n",
       "      <th>57</th>\n",
       "      <th>58</th>\n",
       "      <th>59</th>\n",
       "      <th>60</th>\n",
       "      <th>61</th>\n",
       "      <th>62</th>\n",
       "      <th>63</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000450305</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000456328</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000488147</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td>10/10</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619216</td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00000000002</td>\n",
       "      <td>open_chromatin_region</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>;BaseQRankSum=0;MQRankSum=0;ReadPosRankSum=0;D...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  0   \\\n",
       "0  ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...   \n",
       "0                                                  A   \n",
       "0                                                  A   \n",
       "0                                                  A   \n",
       "0                                                  A   \n",
       "\n",
       "                                             1         2          3   \\\n",
       "0                       downstream_gene_variant  MODIFIER    DDX11L1   \n",
       "0                       downstream_gene_variant  MODIFIER    DDX11L1   \n",
       "0  intron_variant&non_coding_transcript_variant  MODIFIER     WASH7P   \n",
       "0                       downstream_gene_variant  MODIFIER  MIR6859-1   \n",
       "0                     regulatory_region_variant  MODIFIER              \n",
       "\n",
       "                4                  5                6   \\\n",
       "0  ENSG00000223972         Transcript  ENST00000450305   \n",
       "0  ENSG00000223972         Transcript  ENST00000456328   \n",
       "0  ENSG00000227232         Transcript  ENST00000488147   \n",
       "0  ENSG00000278267         Transcript  ENST00000619216   \n",
       "0                   RegulatoryFeature  ENSR00000000002   \n",
       "\n",
       "                                   7  8      9   ... 54 55 56 57 58 59 60 61  \\\n",
       "0  transcribed_unprocessed_pseudogene            ...                           \n",
       "0                processed_transcript            ...                           \n",
       "0              unprocessed_pseudogene     10/10  ...                           \n",
       "0                               miRNA            ...                           \n",
       "0               open_chromatin_region            ...                           \n",
       "\n",
       "  62                                                 63  \n",
       "0                                                        \n",
       "0                                                        \n",
       "0                                                        \n",
       "0                                                        \n",
       "0     ;BaseQRankSum=0;MQRankSum=0;ReadPosRankSum=0;D...  \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To add column names to splitted INFO columns\n",
    "k.columns = [\"Allele\",\"Consequence\",\"IMPACT\",\"SYMBOL\",\"Gene\",\"Feature_type\",\"Feature\",\"BIOTYPE\",\"EXON\",\"INTRON\",\"HGVSc\",\"HGVSp\",\"cDNA_position\",\"CDS_position\",\"Protein_position\",\"Amino_acids\",\"Codons\",\"Existing_variation\",\"DISTANCE\",\"STRAND\",\"FLAGS\",\"VARIANT_CLASS\",\"SYMBOL_SOURCE\",\"HGNC_ID\",\"CANONICAL\",\"TSL\",\"APPRIS\",\"CCDS\",\"ENSP\",\"SWISSPROT\",\"TREMBL\",\"UNIPARC\",\"GENE_PHENO\",\"SIFT\",\"PolyPhen\",\"DOMAINS\",\"HGVS_OFFSET\",\"AF\",\"AFR_AF\",\"AMR_AF\",\"EAS_AF\",\"EUR_AF\",\"SAS_AF\",\"AA_AF\",\"EA_AF\",\"ExAC_AF\",\"ExAC_Adj_AF\",\"ExAC_AFR_AF\",\"ExAC_AMR_AF\",\"ExAC_EAS_AF\",\"ExAC_FIN_AF\",\"ExAC_NFE_AF\",\"ExAC_OTH_AF\",\"ExAC_SAS_AF\",\"MAX_AF\",\"MAX_AF_POPS\",\"CLIN_SIG\",\"SOMATIC\",\"PHENO\",\"PUBMED\",\"MOTIF_NAME\",\"MOTIF_POS\",\"HIGH_INF_POS\",\"MOTIF_SCORE_CHANGE\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000450305</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000456328</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000488147</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td>10/10</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619216</td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00000000002</td>\n",
       "      <td>open_chromatin_region</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>;BaseQRankSum=0;MQRankSum=0;ReadPosRankSum=0;D...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Allele  \\\n",
       "0  ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...   \n",
       "0                                                  A   \n",
       "0                                                  A   \n",
       "0                                                  A   \n",
       "0                                                  A   \n",
       "\n",
       "                                    Consequence    IMPACT     SYMBOL  \\\n",
       "0                       downstream_gene_variant  MODIFIER    DDX11L1   \n",
       "0                       downstream_gene_variant  MODIFIER    DDX11L1   \n",
       "0  intron_variant&non_coding_transcript_variant  MODIFIER     WASH7P   \n",
       "0                       downstream_gene_variant  MODIFIER  MIR6859-1   \n",
       "0                     regulatory_region_variant  MODIFIER              \n",
       "\n",
       "              Gene       Feature_type          Feature  \\\n",
       "0  ENSG00000223972         Transcript  ENST00000450305   \n",
       "0  ENSG00000223972         Transcript  ENST00000456328   \n",
       "0  ENSG00000227232         Transcript  ENST00000488147   \n",
       "0  ENSG00000278267         Transcript  ENST00000619216   \n",
       "0                   RegulatoryFeature  ENSR00000000002   \n",
       "\n",
       "                              BIOTYPE EXON INTRON  ... MAX_AF MAX_AF_POPS  \\\n",
       "0  transcribed_unprocessed_pseudogene              ...                      \n",
       "0                processed_transcript              ...                      \n",
       "0              unprocessed_pseudogene       10/10  ...                      \n",
       "0                               miRNA              ...                      \n",
       "0               open_chromatin_region              ...                      \n",
       "\n",
       "  CLIN_SIG SOMATIC PHENO PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS  \\\n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "\n",
       "                                  MOTIF_SCORE_CHANGE  \n",
       "0                                                     \n",
       "0                                                     \n",
       "0                                                     \n",
       "0                                                     \n",
       "0  ;BaseQRankSum=0;MQRankSum=0;ReadPosRankSum=0;D...  \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To filter only important columns from splitted INFO column and saved in p variable\n",
    "p = k[[\"Consequence\",\"SYMBOL\",\"Gene\",\"Feature_type\",\"Feature\",\"BIOTYPE\",\"EXON\",\"INTRON\",\"Existing_variation\",\"SIFT\",\"PolyPhen\",\"AF\",\"ExAC_AF\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>ExAC_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000450305</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000456328</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000488147</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td>10/10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619216</td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00000000002</td>\n",
       "      <td>open_chromatin_region</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Consequence     SYMBOL             Gene  \\\n",
       "0                       downstream_gene_variant    DDX11L1  ENSG00000223972   \n",
       "0                       downstream_gene_variant    DDX11L1  ENSG00000223972   \n",
       "0  intron_variant&non_coding_transcript_variant     WASH7P  ENSG00000227232   \n",
       "0                       downstream_gene_variant  MIR6859-1  ENSG00000278267   \n",
       "0                     regulatory_region_variant                               \n",
       "\n",
       "        Feature_type          Feature                             BIOTYPE  \\\n",
       "0         Transcript  ENST00000450305  transcribed_unprocessed_pseudogene   \n",
       "0         Transcript  ENST00000456328                processed_transcript   \n",
       "0         Transcript  ENST00000488147              unprocessed_pseudogene   \n",
       "0         Transcript  ENST00000619216                               miRNA   \n",
       "0  RegulatoryFeature  ENSR00000000002               open_chromatin_region   \n",
       "\n",
       "  EXON INTRON Existing_variation SIFT PolyPhen AF ExAC_AF  \n",
       "0                                                          \n",
       "0                                                          \n",
       "0       10/10                                              \n",
       "0                                                          \n",
       "0                                                          "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "#join splitted INFO column by bar (|) with main vcf file \n",
    "final_INFO_splitted_file = df1.join(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample_Count</th>\n",
       "      <th>INFO</th>\n",
       "      <th>...</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>ExAC_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000450305</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000456328</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000488147</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td>10/10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619216</td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...</td>\n",
       "      <td>...</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00000000002</td>\n",
       "      <td>open_chromatin_region</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM    POS ID REF ALT    QUAL FILTER          FORMAT  Sample_Count  \\\n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "\n",
       "                                                INFO  ...       Feature_type  \\\n",
       "0  ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...  ...         Transcript   \n",
       "0  ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...  ...         Transcript   \n",
       "0  ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...  ...         Transcript   \n",
       "0  ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...  ...         Transcript   \n",
       "0  ExcessHet=3.0103;FS=0;MQ=60;QD=33.16;SOR=2.303...  ...  RegulatoryFeature   \n",
       "\n",
       "           Feature                             BIOTYPE EXON INTRON  \\\n",
       "0  ENST00000450305  transcribed_unprocessed_pseudogene               \n",
       "0  ENST00000456328                processed_transcript               \n",
       "0  ENST00000488147              unprocessed_pseudogene       10/10   \n",
       "0  ENST00000619216                               miRNA               \n",
       "0  ENSR00000000002               open_chromatin_region               \n",
       "\n",
       "  Existing_variation SIFT PolyPhen AF ExAC_AF  \n",
       "0                                              \n",
       "0                                              \n",
       "0                                              \n",
       "0                                              \n",
       "0                                              \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_INFO_splitted_file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "79361523"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(final_INFO_splitted_file.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_INFO_splitted_file.to_csv('Asthma_splitted_mutations.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To remove any extra or unwanted columns from Arranged_cols file\n",
    "removed_columns = final_INFO_splitted_file.drop(['INFO'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "All_Asthma_mutation_unique = removed_columns.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6855669"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(All_Asthma_mutation_unique.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#print output file with all mutations and its consequences\n",
    "All_Asthma_mutation_unique.to_csv('Asthma_final_filtered_mutations.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
