{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import csv\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/home/komal.bhosle/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (22) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv('Asthma_final_filtered_mutations.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method NDFrame.head of          Unnamed: 0                    CHROM      POS ID REF ALT    QUAL  \\\n",
       "0                 0                     chr1    14522  .   G   A  118.64   \n",
       "1                 0                     chr1    14522  .   G   A  118.64   \n",
       "2                 0                     chr1    14522  .   G   A  118.64   \n",
       "3                 0                     chr1    14522  .   G   A  118.64   \n",
       "4                 0                     chr1    14522  .   G   A  118.64   \n",
       "5                 1                     chr1    14590  .   G   A   45.64   \n",
       "6                 1                     chr1    14590  .   G   A   45.64   \n",
       "7                 1                     chr1    14590  .   G   A   45.64   \n",
       "8                 1                     chr1    14590  .   G   A   45.64   \n",
       "9                 2                     chr1    14673  .   G   C   71.32   \n",
       "10                2                     chr1    14673  .   G   C   71.32   \n",
       "11                2                     chr1    14673  .   G   C   71.32   \n",
       "12                2                     chr1    14673  .   G   C   71.32   \n",
       "13                3                     chr1    15015  .   G   C  259.65   \n",
       "14                3                     chr1    15015  .   G   C  259.65   \n",
       "15                3                     chr1    15015  .   G   C  259.65   \n",
       "16                3                     chr1    15015  .   G   C  259.65   \n",
       "17                4                     chr1    17385  .   G   A   57.64   \n",
       "18                4                     chr1    17385  .   G   A   57.64   \n",
       "19                4                     chr1    17385  .   G   A   57.64   \n",
       "20                4                     chr1    17385  .   G   A   57.64   \n",
       "21                5                     chr1    17452  .   C   T   84.64   \n",
       "22                5                     chr1    17452  .   C   T   84.64   \n",
       "23                5                     chr1    17452  .   C   T   84.64   \n",
       "24                5                     chr1    17452  .   C   T   84.64   \n",
       "25                6                     chr1    17538  .   C   A  132.64   \n",
       "26                6                     chr1    17538  .   C   A  132.64   \n",
       "27                6                     chr1    17538  .   C   A  132.64   \n",
       "28                6                     chr1    17538  .   C   A  132.64   \n",
       "29                7                     chr1    17594  .   C   T   37.64   \n",
       "...             ...                      ...      ... ..  ..  ..     ...   \n",
       "6855639     1125686         chrUn_KI270330v1     1457  .   G   A  109.64   \n",
       "6855640     1125686         chrUn_KI270330v1     1457  .   G   A  109.64   \n",
       "6855641     1125686         chrUn_KI270330v1     1457  .   G   A  109.64   \n",
       "6855642     1125687         chrUn_KI270330v1     1458  .   T   C  109.64   \n",
       "6855643     1125687         chrUn_KI270330v1     1458  .   T   C  109.64   \n",
       "6855644     1125687         chrUn_KI270330v1     1458  .   T   C  109.64   \n",
       "6855645     1125688         chrUn_GL000216v2   161956  .   C   T   78.32   \n",
       "6855646     1125689         chrUn_GL000216v2   161962  .   C   A   78.32   \n",
       "6855647     1125690         chrUn_GL000216v2   161967  .   C   T   78.32   \n",
       "6855648     1125691         chrUn_GL000216v2   169639  .   G   T   35.48   \n",
       "6855649     1125692      chr8_KI270819v1_alt    86029  .   C   T   66.32   \n",
       "6855650     1125693      chr8_KI270819v1_alt    90475  .   G   A   31.64   \n",
       "6855651     1125694     chr14_KI270847v1_alt  1497716  .   G   T   35.48   \n",
       "6855652     1125695     chr14_KI270847v1_alt  1497717  .   G   T   35.48   \n",
       "6855653     1125696     chr14_KI270847v1_alt  1497719  .   G   A   35.48   \n",
       "6855654     1125697     chr17_KI270858v1_alt   117873  .   T   C   78.32   \n",
       "6855655     1125698     chr17_KI270858v1_alt   117889  .   C   A   78.32   \n",
       "6855656     1125699     chr19_GL383575v2_alt     8234  .   T   C   76.84   \n",
       "6855657     1125700     chr22_KI270875v1_alt     2816  .   G   A   74.32   \n",
       "6855658     1125701     chr22_KI270875v1_alt     2837  .   T   C   78.32   \n",
       "6855659     1125702     chr22_KI270875v1_alt     2839  .   C   T   78.32   \n",
       "6855660     1125703     chr22_KI270875v1_alt     2840  .   C   T   78.32   \n",
       "6855661     1125704     chr22_KI270875v1_alt    29365  .   G   A  142.64   \n",
       "6855662     1125705     chr22_KI270875v1_alt    29441  .   A   C   93.64   \n",
       "6855663     1125706     chr22_KI270875v1_alt    40061  .   A   G   37.32   \n",
       "6855664     1125707      chr7_KI270899v1_alt    13793  .   A   G   35.48   \n",
       "6855665     1125708      chr7_KI270899v1_alt    13797  .   A   T   35.48   \n",
       "6855666     1125709      chr6_KI270802v1_alt    71744  .   T   G   65.32   \n",
       "6855667     1125710      chr7_GL383534v2_alt    75942  .   A   G   73.32   \n",
       "6855668     1125711  chr22_KI270732v1_random    36439  .   T   C   56.32   \n",
       "\n",
       "        FILTER          FORMAT  Sample_Count  ...       Feature_type  \\\n",
       "0         PASS  GT:AD:DP:GQ:PL             2  ...         Transcript   \n",
       "1         PASS  GT:AD:DP:GQ:PL             2  ...         Transcript   \n",
       "2         PASS  GT:AD:DP:GQ:PL             2  ...         Transcript   \n",
       "3         PASS  GT:AD:DP:GQ:PL             2  ...         Transcript   \n",
       "4         PASS  GT:AD:DP:GQ:PL             2  ...  RegulatoryFeature   \n",
       "5         PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "6         PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "7         PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "8         PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "9         PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "10        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "11        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "12        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "13        PASS  GT:AD:DP:GQ:PL             4  ...         Transcript   \n",
       "14        PASS  GT:AD:DP:GQ:PL             4  ...         Transcript   \n",
       "15        PASS  GT:AD:DP:GQ:PL             4  ...         Transcript   \n",
       "16        PASS  GT:AD:DP:GQ:PL             4  ...         Transcript   \n",
       "17        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "18        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "19        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "20        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "21        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "22        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "23        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "24        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "25        PASS  GT:AD:DP:GQ:PL             2  ...         Transcript   \n",
       "26        PASS  GT:AD:DP:GQ:PL             2  ...         Transcript   \n",
       "27        PASS  GT:AD:DP:GQ:PL             2  ...         Transcript   \n",
       "28        PASS  GT:AD:DP:GQ:PL             2  ...         Transcript   \n",
       "29        PASS  GT:AD:DP:GQ:PL             1  ...         Transcript   \n",
       "...        ...             ...           ...  ...                ...   \n",
       "6855639   PASS  GT:AD:DP:GQ:PL             1  ...  RegulatoryFeature   \n",
       "6855640   PASS  GT:AD:DP:GQ:PL             1  ...  RegulatoryFeature   \n",
       "6855641   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855642   PASS  GT:AD:DP:GQ:PL             1  ...  RegulatoryFeature   \n",
       "6855643   PASS  GT:AD:DP:GQ:PL             1  ...  RegulatoryFeature   \n",
       "6855644   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855645   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855646   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855647   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855648   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855649   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855650   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855651   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855652   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855653   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855654   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855655   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855656   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855657   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855658   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855659   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855660   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855661   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855662   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855663   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855664   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855665   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855666   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855667   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "6855668   PASS  GT:AD:DP:GQ:PL             1  ...                NaN   \n",
       "\n",
       "                 Feature                             BIOTYPE   EXON INTRON  \\\n",
       "0        ENST00000450305  transcribed_unprocessed_pseudogene    NaN    NaN   \n",
       "1        ENST00000456328                processed_transcript    NaN    NaN   \n",
       "2        ENST00000488147              unprocessed_pseudogene    NaN  10/10   \n",
       "3        ENST00000619216                               miRNA    NaN    NaN   \n",
       "4        ENSR00000000002               open_chromatin_region    NaN    NaN   \n",
       "5        ENST00000450305  transcribed_unprocessed_pseudogene    NaN    NaN   \n",
       "6        ENST00000456328                processed_transcript    NaN    NaN   \n",
       "7        ENST00000488147              unprocessed_pseudogene    NaN  10/10   \n",
       "8        ENST00000619216                               miRNA    NaN    NaN   \n",
       "9        ENST00000450305  transcribed_unprocessed_pseudogene    NaN    NaN   \n",
       "10       ENST00000456328                processed_transcript    NaN    NaN   \n",
       "11       ENST00000488147              unprocessed_pseudogene    NaN  10/10   \n",
       "12       ENST00000619216                               miRNA    NaN    NaN   \n",
       "13       ENST00000450305  transcribed_unprocessed_pseudogene    NaN    NaN   \n",
       "14       ENST00000456328                processed_transcript    NaN    NaN   \n",
       "15       ENST00000488147              unprocessed_pseudogene  10/11    NaN   \n",
       "16       ENST00000619216                               miRNA    NaN    NaN   \n",
       "17       ENST00000450305  transcribed_unprocessed_pseudogene    NaN    NaN   \n",
       "18       ENST00000456328                processed_transcript    NaN    NaN   \n",
       "19       ENST00000488147              unprocessed_pseudogene    NaN   5/10   \n",
       "20       ENST00000619216                               miRNA    1/1    NaN   \n",
       "21       ENST00000450305  transcribed_unprocessed_pseudogene    NaN    NaN   \n",
       "22       ENST00000456328                processed_transcript    NaN    NaN   \n",
       "23       ENST00000488147              unprocessed_pseudogene    NaN   5/10   \n",
       "24       ENST00000619216                               miRNA    NaN    NaN   \n",
       "25       ENST00000450305  transcribed_unprocessed_pseudogene    NaN    NaN   \n",
       "26       ENST00000456328                processed_transcript    NaN    NaN   \n",
       "27       ENST00000488147              unprocessed_pseudogene    NaN   5/10   \n",
       "28       ENST00000619216                               miRNA    NaN    NaN   \n",
       "29       ENST00000450305  transcribed_unprocessed_pseudogene    NaN    NaN   \n",
       "...                  ...                                 ...    ...    ...   \n",
       "6855639  ENSR00000338588                   CTCF_binding_site    NaN    NaN   \n",
       "6855640  ENSR00000338589                     TF_binding_site    NaN    NaN   \n",
       "6855641              NaN                                 NaN    NaN    NaN   \n",
       "6855642  ENSR00000338588                   CTCF_binding_site    NaN    NaN   \n",
       "6855643  ENSR00000338589                     TF_binding_site    NaN    NaN   \n",
       "6855644              NaN                                 NaN    NaN    NaN   \n",
       "6855645              NaN                                 NaN    NaN    NaN   \n",
       "6855646              NaN                                 NaN    NaN    NaN   \n",
       "6855647              NaN                                 NaN    NaN    NaN   \n",
       "6855648              NaN                                 NaN    NaN    NaN   \n",
       "6855649              NaN                                 NaN    NaN    NaN   \n",
       "6855650              NaN                                 NaN    NaN    NaN   \n",
       "6855651              NaN                                 NaN    NaN    NaN   \n",
       "6855652              NaN                                 NaN    NaN    NaN   \n",
       "6855653              NaN                                 NaN    NaN    NaN   \n",
       "6855654              NaN                                 NaN    NaN    NaN   \n",
       "6855655              NaN                                 NaN    NaN    NaN   \n",
       "6855656              NaN                                 NaN    NaN    NaN   \n",
       "6855657              NaN                                 NaN    NaN    NaN   \n",
       "6855658              NaN                                 NaN    NaN    NaN   \n",
       "6855659              NaN                                 NaN    NaN    NaN   \n",
       "6855660              NaN                                 NaN    NaN    NaN   \n",
       "6855661              NaN                                 NaN    NaN    NaN   \n",
       "6855662              NaN                                 NaN    NaN    NaN   \n",
       "6855663              NaN                                 NaN    NaN    NaN   \n",
       "6855664              NaN                                 NaN    NaN    NaN   \n",
       "6855665              NaN                                 NaN    NaN    NaN   \n",
       "6855666              NaN                                 NaN    NaN    NaN   \n",
       "6855667              NaN                                 NaN    NaN    NaN   \n",
       "6855668              NaN                                 NaN    NaN    NaN   \n",
       "\n",
       "        Existing_variation SIFT PolyPhen  AF ExAC_AF  \n",
       "0                      NaN  NaN      NaN NaN     NaN  \n",
       "1                      NaN  NaN      NaN NaN     NaN  \n",
       "2                      NaN  NaN      NaN NaN     NaN  \n",
       "3                      NaN  NaN      NaN NaN     NaN  \n",
       "4                      NaN  NaN      NaN NaN     NaN  \n",
       "5                 rs707679  NaN      NaN NaN     NaN  \n",
       "6                 rs707679  NaN      NaN NaN     NaN  \n",
       "7                 rs707679  NaN      NaN NaN     NaN  \n",
       "8                 rs707679  NaN      NaN NaN     NaN  \n",
       "9                   rs4690  NaN      NaN NaN     NaN  \n",
       "10                  rs4690  NaN      NaN NaN     NaN  \n",
       "11                  rs4690  NaN      NaN NaN     NaN  \n",
       "12                  rs4690  NaN      NaN NaN     NaN  \n",
       "13                     NaN  NaN      NaN NaN     NaN  \n",
       "14                     NaN  NaN      NaN NaN     NaN  \n",
       "15                     NaN  NaN      NaN NaN     NaN  \n",
       "16                     NaN  NaN      NaN NaN     NaN  \n",
       "17             rs201535981  NaN      NaN NaN   0.122  \n",
       "18             rs201535981  NaN      NaN NaN   0.122  \n",
       "19             rs201535981  NaN      NaN NaN   0.122  \n",
       "20             rs201535981  NaN      NaN NaN   0.122  \n",
       "21             rs200503540  NaN      NaN NaN   0.016  \n",
       "22             rs200503540  NaN      NaN NaN   0.016  \n",
       "23             rs200503540  NaN      NaN NaN   0.016  \n",
       "24             rs200503540  NaN      NaN NaN   0.016  \n",
       "25              rs71260068  NaN      NaN NaN     NaN  \n",
       "26              rs71260068  NaN      NaN NaN     NaN  \n",
       "27              rs71260068  NaN      NaN NaN     NaN  \n",
       "28              rs71260068  NaN      NaN NaN     NaN  \n",
       "29             rs377698370  NaN      NaN NaN     NaN  \n",
       "...                    ...  ...      ...  ..     ...  \n",
       "6855639                NaN  NaN      NaN NaN     NaN  \n",
       "6855640                NaN  NaN      NaN NaN     NaN  \n",
       "6855641                NaN  NaN      NaN NaN     NaN  \n",
       "6855642                NaN  NaN      NaN NaN     NaN  \n",
       "6855643                NaN  NaN      NaN NaN     NaN  \n",
       "6855644                NaN  NaN      NaN NaN     NaN  \n",
       "6855645                NaN  NaN      NaN NaN     NaN  \n",
       "6855646                NaN  NaN      NaN NaN     NaN  \n",
       "6855647                NaN  NaN      NaN NaN     NaN  \n",
       "6855648                NaN  NaN      NaN NaN     NaN  \n",
       "6855649                NaN  NaN      NaN NaN     NaN  \n",
       "6855650                NaN  NaN      NaN NaN     NaN  \n",
       "6855651                NaN  NaN      NaN NaN     NaN  \n",
       "6855652                NaN  NaN      NaN NaN     NaN  \n",
       "6855653                NaN  NaN      NaN NaN     NaN  \n",
       "6855654                NaN  NaN      NaN NaN     NaN  \n",
       "6855655                NaN  NaN      NaN NaN     NaN  \n",
       "6855656                NaN  NaN      NaN NaN     NaN  \n",
       "6855657                NaN  NaN      NaN NaN     NaN  \n",
       "6855658                NaN  NaN      NaN NaN     NaN  \n",
       "6855659                NaN  NaN      NaN NaN     NaN  \n",
       "6855660                NaN  NaN      NaN NaN     NaN  \n",
       "6855661                NaN  NaN      NaN NaN     NaN  \n",
       "6855662                NaN  NaN      NaN NaN     NaN  \n",
       "6855663                NaN  NaN      NaN NaN     NaN  \n",
       "6855664                NaN  NaN      NaN NaN     NaN  \n",
       "6855665                NaN  NaN      NaN NaN     NaN  \n",
       "6855666                NaN  NaN      NaN NaN     NaN  \n",
       "6855667                NaN  NaN      NaN NaN     NaN  \n",
       "6855668                NaN  NaN      NaN NaN     NaN  \n",
       "\n",
       "[6855669 rows x 23 columns]>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To remove other columns\n",
    "removed_columns = df.drop(['Unnamed: 0'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample_Count</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>ExAC_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000450305</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000456328</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000488147</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619216</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>118.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00000000002</td>\n",
       "      <td>open_chromatin_region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM    POS ID REF ALT    QUAL FILTER          FORMAT  Sample_Count  \\\n",
       "0  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "1  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "2  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "3  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "4  chr1  14522  .   G   A  118.64   PASS  GT:AD:DP:GQ:PL             2   \n",
       "\n",
       "                                    Consequence  ...       Feature_type  \\\n",
       "0                       downstream_gene_variant  ...         Transcript   \n",
       "1                       downstream_gene_variant  ...         Transcript   \n",
       "2  intron_variant&non_coding_transcript_variant  ...         Transcript   \n",
       "3                       downstream_gene_variant  ...         Transcript   \n",
       "4                     regulatory_region_variant  ...  RegulatoryFeature   \n",
       "\n",
       "           Feature                             BIOTYPE EXON INTRON  \\\n",
       "0  ENST00000450305  transcribed_unprocessed_pseudogene  NaN    NaN   \n",
       "1  ENST00000456328                processed_transcript  NaN    NaN   \n",
       "2  ENST00000488147              unprocessed_pseudogene  NaN  10/10   \n",
       "3  ENST00000619216                               miRNA  NaN    NaN   \n",
       "4  ENSR00000000002               open_chromatin_region  NaN    NaN   \n",
       "\n",
       "  Existing_variation SIFT PolyPhen  AF ExAC_AF  \n",
       "0                NaN  NaN      NaN NaN     NaN  \n",
       "1                NaN  NaN      NaN NaN     NaN  \n",
       "2                NaN  NaN      NaN NaN     NaN  \n",
       "3                NaN  NaN      NaN NaN     NaN  \n",
       "4                NaN  NaN      NaN NaN     NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "removed_columns.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6855669"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(removed_columns.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Print Missesse variants and na=False for empty columns\n",
    "missense_variants = removed_columns[removed_columns['Consequence'].str.contains('missense_variant', na=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample_Count</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>ExAC_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1998</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.759)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1999</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>14/14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.876)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2001</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>7/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.999)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2010</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.987)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2011</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>13/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.959)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM     POS ID REF ALT   QUAL FILTER          FORMAT  Sample_Count  \\\n",
       "1998  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "1999  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "2001  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "2010  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "2011  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "\n",
       "           Consequence  ... Feature_type          Feature         BIOTYPE  \\\n",
       "1998  missense_variant  ...   Transcript  ENST00000341065  protein_coding   \n",
       "1999  missense_variant  ...   Transcript  ENST00000342066  protein_coding   \n",
       "2001  missense_variant  ...   Transcript  ENST00000455979  protein_coding   \n",
       "2010  missense_variant  ...   Transcript  ENST00000616125  protein_coding   \n",
       "2011  missense_variant  ...   Transcript  ENST00000617307  protein_coding   \n",
       "\n",
       "       EXON INTRON Existing_variation            SIFT  \\\n",
       "1998  12/12    NaN                NaN  deleterious(0)   \n",
       "1999  14/14    NaN                NaN  deleterious(0)   \n",
       "2001    7/7    NaN                NaN  deleterious(0)   \n",
       "2010  12/12    NaN                NaN  deleterious(0)   \n",
       "2011  13/13    NaN                NaN  deleterious(0)   \n",
       "\n",
       "                      PolyPhen  AF ExAC_AF  \n",
       "1998  possibly_damaging(0.759) NaN     NaN  \n",
       "1999  possibly_damaging(0.876) NaN     NaN  \n",
       "2001  probably_damaging(0.999) NaN     NaN  \n",
       "2010  probably_damaging(0.987) NaN     NaN  \n",
       "2011  probably_damaging(0.959) NaN     NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missense_variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "114467"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#to count no of missense variants\n",
    "len(missense_variants.axes[0]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To filter LOF mutations: missense varinats having \"deleterious\" effect in SIFT and \"damaging\" effect in polyphen database\n",
    "LOF_mutations = removed_columns[removed_columns['Consequence'].str.contains(\"missense_variant\") & removed_columns['SIFT'].str.contains(\"deleterious\") & removed_columns['PolyPhen'].str.contains(\"damaging\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample_Count</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>ExAC_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1998</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.759)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1999</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>14/14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.876)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2001</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>7/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.999)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2010</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.987)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2011</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943923</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>70.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>13/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.959)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM     POS ID REF ALT   QUAL FILTER          FORMAT  Sample_Count  \\\n",
       "1998  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "1999  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "2001  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "2010  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "2011  chr1  943923  .   G   A  70.64   PASS  GT:AD:DP:GQ:PL             1   \n",
       "\n",
       "           Consequence  ... Feature_type          Feature         BIOTYPE  \\\n",
       "1998  missense_variant  ...   Transcript  ENST00000341065  protein_coding   \n",
       "1999  missense_variant  ...   Transcript  ENST00000342066  protein_coding   \n",
       "2001  missense_variant  ...   Transcript  ENST00000455979  protein_coding   \n",
       "2010  missense_variant  ...   Transcript  ENST00000616125  protein_coding   \n",
       "2011  missense_variant  ...   Transcript  ENST00000617307  protein_coding   \n",
       "\n",
       "       EXON INTRON Existing_variation            SIFT  \\\n",
       "1998  12/12    NaN                NaN  deleterious(0)   \n",
       "1999  14/14    NaN                NaN  deleterious(0)   \n",
       "2001    7/7    NaN                NaN  deleterious(0)   \n",
       "2010  12/12    NaN                NaN  deleterious(0)   \n",
       "2011  13/13    NaN                NaN  deleterious(0)   \n",
       "\n",
       "                      PolyPhen  AF ExAC_AF  \n",
       "1998  possibly_damaging(0.759) NaN     NaN  \n",
       "1999  possibly_damaging(0.876) NaN     NaN  \n",
       "2001  probably_damaging(0.999) NaN     NaN  \n",
       "2010  probably_damaging(0.987) NaN     NaN  \n",
       "2011  probably_damaging(0.959) NaN     NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LOF_mutations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41691"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#To count no of LOF mutations\n",
    "len(LOF_mutations.axes[0]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#print output\n",
    "LOF_mutations.to_csv('Asthma_LOF_mutations.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Done with LOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Started with other filteration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To print protein coding mutations\n",
    "Protein_coding = removed_columns[removed_columns['BIOTYPE'].str.contains('protein_coding', na=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample_Count</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>ExAC_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1917</th>\n",
       "      <td>chr1</td>\n",
       "      <td>920949</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>66.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs7414599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1825</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1919</th>\n",
       "      <td>chr1</td>\n",
       "      <td>920949</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>66.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000420190</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs7414599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1825</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1921</th>\n",
       "      <td>chr1</td>\n",
       "      <td>920949</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>66.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs7414599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1825</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1924</th>\n",
       "      <td>chr1</td>\n",
       "      <td>920949</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>66.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs7414599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1825</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1925</th>\n",
       "      <td>chr1</td>\n",
       "      <td>920949</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>66.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs7414599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1825</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM     POS ID REF ALT   QUAL FILTER          FORMAT  Sample_Count  \\\n",
       "1917  chr1  920949  .   C   G  66.32   PASS  GT:AD:DP:GQ:PL             1   \n",
       "1919  chr1  920949  .   C   G  66.32   PASS  GT:AD:DP:GQ:PL             1   \n",
       "1921  chr1  920949  .   C   G  66.32   PASS  GT:AD:DP:GQ:PL             1   \n",
       "1924  chr1  920949  .   C   G  66.32   PASS  GT:AD:DP:GQ:PL             1   \n",
       "1925  chr1  920949  .   C   G  66.32   PASS  GT:AD:DP:GQ:PL             1   \n",
       "\n",
       "                Consequence  ... Feature_type          Feature  \\\n",
       "1917  upstream_gene_variant  ...   Transcript  ENST00000342066   \n",
       "1919  upstream_gene_variant  ...   Transcript  ENST00000420190   \n",
       "1921  upstream_gene_variant  ...   Transcript  ENST00000437963   \n",
       "1924  upstream_gene_variant  ...   Transcript  ENST00000616016   \n",
       "1925  upstream_gene_variant  ...   Transcript  ENST00000616125   \n",
       "\n",
       "             BIOTYPE EXON INTRON Existing_variation SIFT PolyPhen      AF  \\\n",
       "1917  protein_coding  NaN    NaN          rs7414599  NaN      NaN  0.1825   \n",
       "1919  protein_coding  NaN    NaN          rs7414599  NaN      NaN  0.1825   \n",
       "1921  protein_coding  NaN    NaN          rs7414599  NaN      NaN  0.1825   \n",
       "1924  protein_coding  NaN    NaN          rs7414599  NaN      NaN  0.1825   \n",
       "1925  protein_coding  NaN    NaN          rs7414599  NaN      NaN  0.1825   \n",
       "\n",
       "     ExAC_AF  \n",
       "1917     NaN  \n",
       "1919     NaN  \n",
       "1921     NaN  \n",
       "1924     NaN  \n",
       "1925     NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Protein_coding.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3875529"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Protein_coding.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Gene symbol with mutation count\n",
    "gene_mutation_count = removed_columns['SYMBOL'].value_counts() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/home/komal.bhosle/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "#print output\n",
    "#gene_mutation_count.to_csv('Asthma_Gene_Mutation_count.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Gene symbol with mutation count\n",
    "df4 = removed_columns[['SYMBOL','BIOTYPE']].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>BIOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>processed_transcript</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    SYMBOL                             BIOTYPE\n",
       "0  DDX11L1  transcribed_unprocessed_pseudogene\n",
       "1  DDX11L1                processed_transcript"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df4.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To put filter on allele frequency\n",
    "df5 = removed_columns[removed_columns['AF'] < 0.01]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample_Count</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>ExAC_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>122872</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>37.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000466430</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs62642125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2356</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>122872</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>37.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000471248</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/2</td>\n",
       "      <td>rs62642125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2356</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>122872</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>37.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000477740</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/3</td>\n",
       "      <td>rs62642125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2356</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>122872</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>37.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000610542</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/3</td>\n",
       "      <td>rs62642125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2356</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>2</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs374499151</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0389</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM     POS ID REF ALT    QUAL FILTER          FORMAT  Sample_Count  \\\n",
       "41  chr1  122872  .   T   G   37.32   PASS  GT:AD:DP:GQ:PL             1   \n",
       "42  chr1  122872  .   T   G   37.32   PASS  GT:AD:DP:GQ:PL             1   \n",
       "43  chr1  122872  .   T   G   37.32   PASS  GT:AD:DP:GQ:PL             1   \n",
       "44  chr1  122872  .   T   G   37.32   PASS  GT:AD:DP:GQ:PL             1   \n",
       "45  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL             2   \n",
       "\n",
       "                                     Consequence  ... Feature_type  \\\n",
       "41                         upstream_gene_variant  ...   Transcript   \n",
       "42  intron_variant&non_coding_transcript_variant  ...   Transcript   \n",
       "43  intron_variant&non_coding_transcript_variant  ...   Transcript   \n",
       "44  intron_variant&non_coding_transcript_variant  ...   Transcript   \n",
       "45                       downstream_gene_variant  ...   Transcript   \n",
       "\n",
       "            Feature               BIOTYPE EXON INTRON Existing_variation SIFT  \\\n",
       "41  ENST00000466430               lincRNA  NaN    NaN         rs62642125  NaN   \n",
       "42  ENST00000471248               lincRNA  NaN    1/2         rs62642125  NaN   \n",
       "43  ENST00000477740               lincRNA  NaN    1/3         rs62642125  NaN   \n",
       "44  ENST00000610542               lincRNA  NaN    2/3         rs62642125  NaN   \n",
       "45  ENST00000442987  processed_pseudogene  NaN    NaN        rs374499151  NaN   \n",
       "\n",
       "   PolyPhen      AF ExAC_AF  \n",
       "41      NaN  0.2356     NaN  \n",
       "42      NaN  0.2356     NaN  \n",
       "43      NaN  0.2356     NaN  \n",
       "44      NaN  0.2356     NaN  \n",
       "45      NaN  0.0389     NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df5.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2908184"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df5.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "639642"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df5.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'<' not supported between instances of 'str' and 'float'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-54-951a9010f8ae>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdf6\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mremoved_columns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mremoved_columns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'AF'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m0.01\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m&\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mremoved_columns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'ExAC_AF'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m0.01\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/ops.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(self, other, axis)\u001b[0m\n\u001b[1;32m   1764\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1765\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merrstate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mall\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'ignore'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1766\u001b[0;31m                 \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mna_op\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1767\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mres\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1768\u001b[0m                 raise TypeError('Could not compare {typ} type with Series'\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/ops.py\u001b[0m in \u001b[0;36mna_op\u001b[0;34m(x, y)\u001b[0m\n\u001b[1;32m   1623\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1624\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mis_object_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1625\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_comp_method_OBJECT_ARRAY\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mop\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1626\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1627\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mis_datetimelike_v_numeric\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/ops.py\u001b[0m in \u001b[0;36m_comp_method_OBJECT_ARRAY\u001b[0;34m(op, x, y)\u001b[0m\n\u001b[1;32m   1601\u001b[0m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlibops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvec_compare\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1602\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1603\u001b[0;31m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlibops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscalar_compare\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1604\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1605\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/ops.pyx\u001b[0m in \u001b[0;36mpandas._libs.ops.scalar_compare\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: '<' not supported between instances of 'str' and 'float'"
     ]
    }
   ],
   "source": [
    "df6 = removed_columns[[removed_columns['AF'] < 0.01] & [removed_columns['ExAC_AF'] < 0.01]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
