{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('Info_column_splited_Asthma_variants.txt',sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method NDFrame.head of                           CHROM      POS ID REF  ALT    QUAL  \\\n",
       "0                          chr1   135163  .   C    T  105.84   \n",
       "1                          chr1   135195  .   A    G  102.64   \n",
       "2                          chr1   629190  .   C    T  194.96   \n",
       "3                          chr1   629241  .   C    T  199.96   \n",
       "4                          chr1   629277  .   C    T   69.32   \n",
       "5                          chr1   629885  .   A    G   95.84   \n",
       "6                          chr1   629912  .   T    C   43.64   \n",
       "7                          chr1   630020  .   C    T   74.32   \n",
       "8                          chr1   630174  .   T    C  242.97   \n",
       "9                          chr1   630489  .   A  C,G  110.84   \n",
       "10                         chr1   630496  .   C    T   64.64   \n",
       "11                         chr1   630668  .   G    A   33.48   \n",
       "12                         chr1   630669  .   C    A   33.48   \n",
       "13                         chr1   631385  .   C    A   55.64   \n",
       "14                         chr1   631386  .   T    G   55.64   \n",
       "15                         chr1   631388  .   A    T   55.64   \n",
       "16                         chr1   631622  .   T    C  172.64   \n",
       "17                         chr1   631625  .   C    T  297.02   \n",
       "18                         chr1   631861  .   G    A   36.64   \n",
       "19                         chr1   632429  .   T    C   73.32   \n",
       "20                         chr1   632695  .   T  C,A   78.32   \n",
       "21                         chr1   632984  .   C    T  411.06   \n",
       "22                         chr1   634435  .   G    A   73.32   \n",
       "23                         chr1   634696  .   T  *,C   79.04   \n",
       "24                         chr1   634700  .   A    T  117.31   \n",
       "25                         chr1   634701  .   C    A  112.64   \n",
       "26                         chr1   634709  .   C  T,A  265.10   \n",
       "27                         chr1   634710  .   T    A   31.64   \n",
       "28                         chr1   736733  .   C    T   64.32   \n",
       "29                         chr1   818025  .   C    A   73.32   \n",
       "...                         ...      ... ..  ..  ...     ...   \n",
       "134749         chrUn_KI270742v1     1386  .   G    A   67.32   \n",
       "134750         chrUn_KI270742v1     9659  .   A    T   74.32   \n",
       "134751      chr8_KI270816v1_alt   107290  .   G    A   74.32   \n",
       "134752     chr10_GL383545v1_alt    54078  .   T    C   69.32   \n",
       "134753      chr7_KI270803v1_alt   524291  .   C    G   78.32   \n",
       "134754      chr7_KI270803v1_alt   524293  .   G    T   78.32   \n",
       "134755      chr7_KI270803v1_alt   578762  .   C    T   74.32   \n",
       "134756      chr7_KI270803v1_alt   825051  .   A    G   74.32   \n",
       "134757     chr16_KI270854v1_alt    89399  .   G    C   74.32   \n",
       "134758      chr5_KI270792v1_alt    79087  .   G    A  114.84   \n",
       "134759     chr11_KI270830v1_alt    14726  .   T    C   74.32   \n",
       "134760  chr16_KI270728v1_random  1230477  .   A    G   57.32   \n",
       "134761  chr14_GL000009v2_random   115247  .   A    C   74.32   \n",
       "134762     chr11_KI270832v1_alt    81357  .   G    A   65.64   \n",
       "134763      chr3_KI270936v1_alt    69137  .   T    G   65.32   \n",
       "134764      chr3_KI270936v1_alt    69182  .   G    A  110.84   \n",
       "134765      chr6_GL383533v1_alt    89261  .   G    C   78.32   \n",
       "134766      chr6_GL383533v1_alt    89269  .   T    G   78.32   \n",
       "134767     chr19_GL949752v1_alt   178675  .   A    G   69.64   \n",
       "134768     chr19_GL949752v1_alt   179492  .   C    T   53.64   \n",
       "134769     chr17_GL000258v2_alt  1151098  .   A    T   73.32   \n",
       "134770     chr17_GL000258v2_alt  1397557  .   T    C  195.96   \n",
       "134771     chr17_KI270860v1_alt    18697  .   C    T   62.64   \n",
       "134772     chr17_KI270857v1_alt   304330  .   T    A  113.84   \n",
       "134773     chr17_KI270857v1_alt   329661  .   C    T  103.84   \n",
       "134774     chr17_KI270857v1_alt   353136  .   A    G   74.32   \n",
       "134775     chr17_KI270857v1_alt  1645091  .   G    C   65.32   \n",
       "134776     chr13_KI270840v1_alt   175227  .   C    G   74.32   \n",
       "134777      chr2_KI270772v1_alt    38345  .   A    G   74.32   \n",
       "134778     chr17_KI270908v1_alt  1083520  .   A    G   74.32   \n",
       "\n",
       "                  FILTER  Unnamed: 7  \\\n",
       "0                   PASS         NaN   \n",
       "1                   PASS         NaN   \n",
       "2                   PASS         NaN   \n",
       "3                   PASS         NaN   \n",
       "4                   PASS         NaN   \n",
       "5                   PASS         NaN   \n",
       "6                   PASS         NaN   \n",
       "7                   PASS         NaN   \n",
       "8                   PASS         NaN   \n",
       "9                   PASS         NaN   \n",
       "10                  PASS         NaN   \n",
       "11                  PASS         NaN   \n",
       "12                  PASS         NaN   \n",
       "13                  PASS         NaN   \n",
       "14                  PASS         NaN   \n",
       "15                  PASS         NaN   \n",
       "16                  PASS         NaN   \n",
       "17                  PASS         NaN   \n",
       "18      basic_SNP_filter         NaN   \n",
       "19                  PASS         NaN   \n",
       "20                  PASS         NaN   \n",
       "21                  PASS         NaN   \n",
       "22                  PASS         NaN   \n",
       "23                  PASS         NaN   \n",
       "24                  PASS         NaN   \n",
       "25                  PASS         NaN   \n",
       "26                  PASS         NaN   \n",
       "27                  PASS         NaN   \n",
       "28                  PASS         NaN   \n",
       "29                  PASS         NaN   \n",
       "...                  ...         ...   \n",
       "134749              PASS         NaN   \n",
       "134750              PASS         NaN   \n",
       "134751              PASS         NaN   \n",
       "134752              PASS         NaN   \n",
       "134753              PASS         NaN   \n",
       "134754              PASS         NaN   \n",
       "134755              PASS         NaN   \n",
       "134756              PASS         NaN   \n",
       "134757              PASS         NaN   \n",
       "134758              PASS         NaN   \n",
       "134759              PASS         NaN   \n",
       "134760              PASS         NaN   \n",
       "134761              PASS         NaN   \n",
       "134762              PASS         NaN   \n",
       "134763              PASS         NaN   \n",
       "134764              PASS         NaN   \n",
       "134765              PASS         NaN   \n",
       "134766              PASS         NaN   \n",
       "134767              PASS         NaN   \n",
       "134768              PASS         NaN   \n",
       "134769              PASS         NaN   \n",
       "134770              PASS         NaN   \n",
       "134771              PASS         NaN   \n",
       "134772              PASS         NaN   \n",
       "134773              PASS         NaN   \n",
       "134774              PASS         NaN   \n",
       "134775              PASS         NaN   \n",
       "134776              PASS         NaN   \n",
       "134777              PASS         NaN   \n",
       "134778              PASS         NaN   \n",
       "\n",
       "                                                     INFO          FORMAT  \\\n",
       "0       BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  GT:AD:DP:GQ:PL   \n",
       "1       BaseQRankSum=-0.524;ExcessHet=3.0103;FS=0;MQ=6...  GT:AD:DP:GQ:PL   \n",
       "2       ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=3.611...  GT:AD:DP:GQ:PL   \n",
       "3       ExcessHet=3.0103;FS=0;MQ=60;QD=34.95;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "4       ExcessHet=3.0103;FS=0;MQ=60;QD=34.66;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "5       ExcessHet=3.0103;FS=0;MQ=60;QD=31.95;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "6       BaseQRankSum=-0.423;ExcessHet=3.0103;FS=0;MQ=6...  GT:AD:DP:GQ:PL   \n",
       "7       ExcessHet=3.0103;FS=0;MQ=60;QD=28.73;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "8       ExcessHet=3.0103;FS=0;MQ=60;QD=28.73;SOR=3.912...  GT:AD:DP:GQ:PL   \n",
       "9       BaseQRankSum=0.842;ExcessHet=3.0103;FS=0;MQ=60...  GT:AD:DP:GQ:PL   \n",
       "10      BaseQRankSum=0.126;ExcessHet=3.0103;FS=0;MQ=60...  GT:AD:DP:GQ:PL   \n",
       "11      ExcessHet=3.0103;FS=0;MQ=60;QD=33.48;SOR=1.609...  GT:AD:DP:GQ:PL   \n",
       "12      ExcessHet=3.0103;FS=0;MQ=60;QD=33.48;SOR=1.609...  GT:AD:DP:GQ:PL   \n",
       "13      BaseQRankSum=0.282;ExcessHet=3.0103;FS=0;MQ=60...  GT:AD:DP:GQ:PL   \n",
       "14      BaseQRankSum=0.282;ExcessHet=3.0103;FS=0;MQ=60...  GT:AD:DP:GQ:PL   \n",
       "15      BaseQRankSum=-0.967;ExcessHet=3.0103;FS=0;MQ=6...  GT:AD:DP:GQ:PL   \n",
       "16      BaseQRankSum=-0.366;ExcessHet=3.0103;FS=0;MQ=6...  GT:AD:DP:GQ:PL   \n",
       "17      ExcessHet=3.0103;FS=0;MQ=60;QD=30.97;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "18      BaseQRankSum=0.582;ExcessHet=3.0103;FS=0;MQ=60...  GT:AD:DP:GQ:PL   \n",
       "19      ExcessHet=3.0103;FS=0;MQ=60;QD=28.2;SOR=2.303;...  GT:AD:DP:GQ:PL   \n",
       "20      ExcessHet=3.0103;FS=0;MQ=60;QD=28.2;SOR=2.303;...  GT:AD:DP:GQ:PL   \n",
       "21      ExcessHet=3.0103;FS=0;MQ=60;QD=27.28;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "22      ExcessHet=3.0103;FS=0;MQ=60;QD=30.97;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "23      BaseQRankSum=1.15;ExcessHet=3.0103;FS=0;MQ=60;...  GT:AD:DP:GQ:PL   \n",
       "24      ExcessHet=3.0103;FS=0;MQ=60;MQRankSum=0;QD=29....  GT:AD:DP:GQ:PL   \n",
       "25      ExcessHet=3.0103;FS=0;MQ=60;MQRankSum=0;QD=22....  GT:AD:DP:GQ:PL   \n",
       "26      BaseQRankSum=1.834;ExcessHet=3.0103;FS=0;MQ=60...  GT:AD:DP:GQ:PL   \n",
       "27      BaseQRankSum=-1.981;ExcessHet=3.0103;FS=0;MQ=6...  GT:AD:DP:GQ:PL   \n",
       "28      ExcessHet=3.0103;FS=0;MQ=60;QD=32.16;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "29      ExcessHet=3.0103;FS=0;MQ=60;QD=28.73;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "...                                                   ...             ...   \n",
       "134749  ExcessHet=3.0103;FS=0;MQ=60;QD=33.66;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134750  ExcessHet=3.0103;FS=0;MQ=60;QD=32.21;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134751  ExcessHet=3.0103;FS=0;MQ=60;QD=26.91;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134752  ExcessHet=3.0103;FS=0;MQ=60;QD=34.66;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134753  ExcessHet=3.0103;FS=0;MQ=60;QD=25.92;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134754  ExcessHet=3.0103;FS=0;MQ=60;QD=30.33;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134755  ExcessHet=3.0103;FS=0;MQ=60;QD=28.85;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134756  ExcessHet=3.0103;FS=0;MQ=60;QD=35.39;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134757  ExcessHet=3.0103;FS=0;MQ=60;QD=33.74;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134758  ExcessHet=3.0103;FS=0;MQ=60;QD=26.76;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "134759  ExcessHet=3.0103;FS=0;MQ=60;QD=31.43;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134760  ExcessHet=3.0103;FS=0;MQ=60;QD=28.66;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134761  ExcessHet=3.0103;FS=0;MQ=60;QD=26.47;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134762  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  GT:AD:DP:GQ:PL   \n",
       "134763  ExcessHet=3.0103;FS=0;MQ=60;QD=32.66;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134764  ExcessHet=3.0103;FS=0;MQ=60;QD=33.71;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "134765  ExcessHet=3.0103;FS=0;MQ=60;QD=33.06;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134766  ExcessHet=3.0103;FS=0;MQ=60;QD=33.51;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134767  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  GT:AD:DP:GQ:PL   \n",
       "134768  BaseQRankSum=0.619;ExcessHet=3.0103;FS=0;MQ=60...  GT:AD:DP:GQ:PL   \n",
       "134769  ExcessHet=3.0103;FS=0;MQ=60;QD=33.64;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134770  ExcessHet=3.0103;FS=0;MQ=60;QD=27.88;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134771  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  GT:AD:DP:GQ:PL   \n",
       "134772  ExcessHet=3.0103;FS=0;MQ=60;QD=32.38;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "134773  ExcessHet=3.0103;FS=0;MQ=60;QD=34.61;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "134774  ExcessHet=3.0103;FS=0;MQ=60;QD=32.49;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134775  ExcessHet=3.0103;FS=0;MQ=60;QD=32.66;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134776  ExcessHet=3.0103;FS=0;MQ=60;QD=24.13;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134777  ExcessHet=3.0103;FS=0;MQ=60;QD=32.92;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "134778  ExcessHet=3.0103;FS=0;MQ=60;QD=26.06;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "\n",
       "        ...          SRR7721278               SRR7721279  \\\n",
       "0       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "1       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "2       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "3       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "4       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "5       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "6       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "7       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "8       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "9       ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "10      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "11      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "12      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "13      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "14      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "15      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "16      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "17      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "18      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "19      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "20      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "21      ...  1/1:0,3:3:9:95,9,0  1/1:0,11:11:33:425,33,0   \n",
       "22      ...         ./.:.:.:.:.       1/1:0,2:2:6:85,6,0   \n",
       "23      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "24      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "25      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "26      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "27      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "28      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "29      ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "...     ...                 ...                      ...   \n",
       "134749  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134750  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134751  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134752  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134753  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134754  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134755  ...         ./.:.:.:.:.       1/1:0,2:2:6:86,6,0   \n",
       "134756  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134757  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134758  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134759  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134760  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134761  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134762  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134763  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134764  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134765  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134766  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134767  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134768  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134769  ...         ./.:.:.:.:.       1/1:0,2:2:6:85,6,0   \n",
       "134770  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134771  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134772  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134773  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134774  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134775  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134776  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134777  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "134778  ...         ./.:.:.:.:.              ./.:.:.:.:.   \n",
       "\n",
       "                  SRR7721280                   SRR7721281          SRR7721282  \\\n",
       "0       0/1:1,2:3:34:77,0,34          1/1:0,3:3:9:119,9,0         ./.:.:.:.:.   \n",
       "1                ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "2                ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "3                ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "4                ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "5                ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "6                ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "7                ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "8                ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "9                ./.:.:.:.:.  2/2:0,.,3:3:9:124,.,.,9,.,0         ./.:.:.:.:.   \n",
       "10               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "11               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "12               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "13               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "14               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "15               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "16               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "17               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "18               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "19               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "20               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "21               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "22               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "23               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "24               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "25               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "26               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "27               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "28               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "29               ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "...                      ...                          ...                 ...   \n",
       "134749           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134750           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134751           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134752           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134753           ./.:.:.:.:.                  ./.:.:.:.:.  1/1:0,2:2:6:90,6,0   \n",
       "134754           ./.:.:.:.:.                  ./.:.:.:.:.  1/1:0,2:2:6:90,6,0   \n",
       "134755           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134756           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134757           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134758           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134759           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134760           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134761           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134762           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134763           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134764           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134765           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134766           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134767           ./.:.:.:.:.         0/1:1,2:3:34:77,0,34         ./.:.:.:.:.   \n",
       "134768           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134769           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134770           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134771           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134772           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134773           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134774           ./.:.:.:.:.           1/1:0,2:2:6:86,6,0         ./.:.:.:.:.   \n",
       "134775    1/1:0,2:2:6:77,6,0                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134776           ./.:.:.:.:.           1/1:0,2:2:6:86,6,0         ./.:.:.:.:.   \n",
       "134777           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "134778           ./.:.:.:.:.                  ./.:.:.:.:.         ./.:.:.:.:.   \n",
       "\n",
       "                 SRR7721283   SRR7721284           SRR7721285  \\\n",
       "0               ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "1               ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "2               ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "3               ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "4               ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "5               ./.:.:.:.:.  ./.:.:.:.:.  1/1:0,3:3:9:109,9,0   \n",
       "6               ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "7               ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "8               ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "9               ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "10              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "11              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "12              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "13              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "14              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "15              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "16              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "17              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "18              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "19              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "20              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "21              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "22              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "23              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "24              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "25              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "26              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "27              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "28              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "29              ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "...                     ...          ...                  ...   \n",
       "134749          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134750          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134751          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134752          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134753          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134754          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134755          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134756          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134757          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134758          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134759          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134760          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134761          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134762          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134763          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134764          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134765          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134766          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134767          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134768          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134769          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134770          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134771          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134772          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134773  1/1:0,3:3:9:117,9,0  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134774          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134775          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134776          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134777   1/1:0,2:2:6:86,6,0  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "134778          ./.:.:.:.:.  ./.:.:.:.:.          ./.:.:.:.:.   \n",
       "\n",
       "                  SRR7721286           SRR7721287  \n",
       "0                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "1                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "2                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "3                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "4                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "5                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "6                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "7                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "8                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "9                ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "10               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "11               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "12               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "13               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "14               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "15               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "16               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "17               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "18               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "19               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "20               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "21               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "22               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "23               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "24               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "25               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "26               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "27               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "28               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "29               ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "...                      ...                  ...  \n",
       "134749           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134750           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134751           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134752           ./.:.:.:.:.   1/1:0,2:2:6:81,6,0  \n",
       "134753           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134754           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134755           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134756           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134757           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134758           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134759           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134760           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134761           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134762           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134763           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134764           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134765           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134766           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134767           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134768           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134769  0/1:2,2:4:57:73,0,57          ./.:.:.:.:.  \n",
       "134770           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134771           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134772           ./.:.:.:.:.  1/1:0,3:3:9:127,9,0  \n",
       "134773           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134774           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134775           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134776           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134777           ./.:.:.:.:.          ./.:.:.:.:.  \n",
       "134778           ./.:.:.:.:.   1/1:0,2:2:6:86,6,0  \n",
       "\n",
       "[134779 rows x 78 columns]>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...\n",
       "1    BaseQRankSum=-0.524;ExcessHet=3.0103;FS=0;MQ=6...\n",
       "2    ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=3.611...\n",
       "3    ExcessHet=3.0103;FS=0;MQ=60;QD=34.95;SOR=2.833...\n",
       "4    ExcessHet=3.0103;FS=0;MQ=60;QD=34.66;SOR=2.303...\n",
       "Name: INFO, dtype: object"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"INFO\"].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Split INFO column by bar (|) and save it in k variable\n",
    "k = df.INFO.str.split(\"|\",expand=True,)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "64"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#To count no of columns\n",
    "len(k.axes[1]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'k' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-a093a6c08514>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m#To count no of rows\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maxes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'k' is not defined"
     ]
    }
   ],
   "source": [
    "#To count no of rows\n",
    "len(k.axes[0]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To add column names to splitted INFO columns\n",
    "k.columns = [\"Allele\",\"Consequence\",\"IMPACT\",\"SYMBOL\",\"Gene\",\"Feature_type\",\"Feature\",\"BIOTYPE\",\"EXON\",\"INTRON\",\"HGVSc\",\"HGVSp\",\"cDNA_position\",\"CDS_position\",\"Protein_position\",\"Amino_acids\",\"Codons\",\"Existing_variation\",\"DISTANCE\",\"STRAND\",\"FLAGS\",\"VARIANT_CLASS\",\"SYMBOL_SOURCE\",\"HGNC_ID\",\"CANONICAL\",\"TSL\",\"APPRIS\",\"CCDS\",\"ENSP\",\"SWISSPROT\",\"TREMBL\",\"UNIPARC\",\"GENE_PHENO\",\"SIFT\",\"PolyPhen\",\"DOMAINS\",\"HGVS_OFFSET\",\"AF\",\"AFR_AF\",\"AMR_AF\",\"EAS_AF\",\"EUR_AF\",\"SAS_AF\",\"AA_AF\",\"EA_AF\",\"ExAC_AF\",\"ExAC_Adj_AF\",\"ExAC_AFR_AF\",\"ExAC_AMR_AF\",\"ExAC_EAS_AF\",\"ExAC_FIN_AF\",\"ExAC_NFE_AF\",\"ExAC_OTH_AF\",\"ExAC_SAS_AF\",\"MAX_AF\",\"MAX_AF_POPS\",\"CLIN_SIG\",\"SOMATIC\",\"PHENO\",\"PUBMED\",\"MOTIF_NAME\",\"MOTIF_POS\",\"HIGH_INF_POS\",\"MOTIF_SCORE_CHANGE\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BaseQRankSum=-0.524;ExcessHet=3.0103;FS=0;MQ=6...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0149</td>\n",
       "      <td>EUR</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=3.611...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=34.95;SOR=2.833...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=34.66;SOR=2.303...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Allele              Consequence  \\\n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  downstream_gene_variant   \n",
       "1  BaseQRankSum=-0.524;ExcessHet=3.0103;FS=0;MQ=6...  downstream_gene_variant   \n",
       "2  ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=3.611...    upstream_gene_variant   \n",
       "3  ExcessHet=3.0103;FS=0;MQ=60;QD=34.95;SOR=2.833...    upstream_gene_variant   \n",
       "4  ExcessHet=3.0103;FS=0;MQ=60;QD=34.66;SOR=2.303...    upstream_gene_variant   \n",
       "\n",
       "     IMPACT    SYMBOL             Gene Feature_type          Feature  \\\n",
       "0  MODIFIER    CICP27  ENSG00000233750   Transcript  ENST00000442987   \n",
       "1  MODIFIER    CICP27  ENSG00000233750   Transcript  ENST00000442987   \n",
       "2  MODIFIER  MTCO1P12  ENSG00000237973   Transcript  ENST00000414273   \n",
       "3  MODIFIER  MTCO1P12  ENSG00000237973   Transcript  ENST00000414273   \n",
       "4  MODIFIER  MTCO1P12  ENSG00000237973   Transcript  ENST00000414273   \n",
       "\n",
       "                  BIOTYPE EXON INTRON  ...  MAX_AF MAX_AF_POPS CLIN_SIG  \\\n",
       "0    processed_pseudogene              ...  0.0923         EAS            \n",
       "1    processed_pseudogene              ...  0.0149         EUR            \n",
       "2  unprocessed_pseudogene              ...                                \n",
       "3  unprocessed_pseudogene              ...                                \n",
       "4  unprocessed_pseudogene              ...                                \n",
       "\n",
       "  SOMATIC PHENO PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE  \n",
       "0                                                                            \n",
       "1                                                                            \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To filter only important columns from splitted INFO column and saved in p variable\n",
    "p = k[[\"Consequence\",\"SYMBOL\",\"Gene\",\"Feature_type\",\"Feature\",\"BIOTYPE\",\"EXON\",\"INTRON\",\"Existing_variation\",\"SIFT\",\"PolyPhen\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs554762511</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs879055279</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Consequence    SYMBOL             Gene Feature_type  \\\n",
       "0  downstream_gene_variant    CICP27  ENSG00000233750   Transcript   \n",
       "1  downstream_gene_variant    CICP27  ENSG00000233750   Transcript   \n",
       "2    upstream_gene_variant  MTCO1P12  ENSG00000237973   Transcript   \n",
       "3    upstream_gene_variant  MTCO1P12  ENSG00000237973   Transcript   \n",
       "4    upstream_gene_variant  MTCO1P12  ENSG00000237973   Transcript   \n",
       "\n",
       "           Feature                 BIOTYPE EXON INTRON Existing_variation  \\\n",
       "0  ENST00000442987    processed_pseudogene                    rs374499151   \n",
       "1  ENST00000442987    processed_pseudogene                    rs554762511   \n",
       "2  ENST00000414273  unprocessed_pseudogene                    rs879055279   \n",
       "3  ENST00000414273  unprocessed_pseudogene                                  \n",
       "4  ENST00000414273  unprocessed_pseudogene                                  \n",
       "\n",
       "  SIFT PolyPhen  \n",
       "0                \n",
       "1                \n",
       "2                \n",
       "3                \n",
       "4                "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "#join splitted INFO column with main vcf file \n",
    "final_INFO_splitted_file = df.join(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>...</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>...</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135195</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>102.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BaseQRankSum=-0.524;ExcessHet=3.0103;FS=0;MQ=6...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>...</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs554762511</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629190</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>194.96</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=3.611...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>...</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs879055279</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629241</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>199.96</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=34.95;SOR=2.833...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>...</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629277</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>69.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=34.66;SOR=2.303...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>...</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 89 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS ID REF ALT    QUAL FILTER  Unnamed: 7  \\\n",
       "0  chr1  135163  .   C   T  105.84   PASS         NaN   \n",
       "1  chr1  135195  .   A   G  102.64   PASS         NaN   \n",
       "2  chr1  629190  .   C   T  194.96   PASS         NaN   \n",
       "3  chr1  629241  .   C   T  199.96   PASS         NaN   \n",
       "4  chr1  629277  .   C   T   69.32   PASS         NaN   \n",
       "\n",
       "                                                INFO          FORMAT  ...  \\\n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  GT:AD:DP:GQ:PL  ...   \n",
       "1  BaseQRankSum=-0.524;ExcessHet=3.0103;FS=0;MQ=6...  GT:AD:DP:GQ:PL  ...   \n",
       "2  ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=3.611...  GT:AD:DP:GQ:PL  ...   \n",
       "3  ExcessHet=3.0103;FS=0;MQ=60;QD=34.95;SOR=2.833...  GT:AD:DP:GQ:PL  ...   \n",
       "4  ExcessHet=3.0103;FS=0;MQ=60;QD=34.66;SOR=2.303...  GT:AD:DP:GQ:PL  ...   \n",
       "\n",
       "     SYMBOL             Gene Feature_type          Feature  \\\n",
       "0    CICP27  ENSG00000233750   Transcript  ENST00000442987   \n",
       "1    CICP27  ENSG00000233750   Transcript  ENST00000442987   \n",
       "2  MTCO1P12  ENSG00000237973   Transcript  ENST00000414273   \n",
       "3  MTCO1P12  ENSG00000237973   Transcript  ENST00000414273   \n",
       "4  MTCO1P12  ENSG00000237973   Transcript  ENST00000414273   \n",
       "\n",
       "                  BIOTYPE EXON INTRON Existing_variation SIFT PolyPhen  \n",
       "0    processed_pseudogene                    rs374499151                \n",
       "1    processed_pseudogene                    rs554762511                \n",
       "2  unprocessed_pseudogene                    rs879055279                \n",
       "3  unprocessed_pseudogene                                               \n",
       "4  unprocessed_pseudogene                                               \n",
       "\n",
       "[5 rows x 89 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_INFO_splitted_file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To remove any extra or unwanted columns from final_INFO_splitted_file\n",
    "removed_columns = final_INFO_splitted_file.drop(['Unnamed: 7','INFO'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SRR12043174</th>\n",
       "      <th>SRR12043175</th>\n",
       "      <th>...</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135195</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>102.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs554762511</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629190</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>194.96</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs879055279</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629241</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>199.96</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629277</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>69.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>MTCO1P12</td>\n",
       "      <td>ENSG00000237973</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS ID REF ALT    QUAL FILTER          FORMAT  SRR12043174  \\\n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "1  chr1  135195  .   A   G  102.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "2  chr1  629190  .   C   T  194.96   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "3  chr1  629241  .   C   T  199.96   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "4  chr1  629277  .   C   T   69.32   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "\n",
       "   SRR12043175  ...    SYMBOL             Gene Feature_type          Feature  \\\n",
       "0  ./.:.:.:.:.  ...    CICP27  ENSG00000233750   Transcript  ENST00000442987   \n",
       "1  ./.:.:.:.:.  ...    CICP27  ENSG00000233750   Transcript  ENST00000442987   \n",
       "2  ./.:.:.:.:.  ...  MTCO1P12  ENSG00000237973   Transcript  ENST00000414273   \n",
       "3  ./.:.:.:.:.  ...  MTCO1P12  ENSG00000237973   Transcript  ENST00000414273   \n",
       "4  ./.:.:.:.:.  ...  MTCO1P12  ENSG00000237973   Transcript  ENST00000414273   \n",
       "\n",
       "                  BIOTYPE EXON INTRON Existing_variation SIFT PolyPhen  \n",
       "0    processed_pseudogene                    rs374499151                \n",
       "1    processed_pseudogene                    rs554762511                \n",
       "2  unprocessed_pseudogene                    rs879055279                \n",
       "3  unprocessed_pseudogene                                               \n",
       "4  unprocessed_pseudogene                                               \n",
       "\n",
       "[5 rows x 87 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "removed_columns.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "#to check only missesnse variants\n",
    "missense_variants = removed_columns[removed_columns['Consequence'].str.contains(\"missense_variant\")] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SRR12043174</th>\n",
       "      <th>SRR12043175</th>\n",
       "      <th>...</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945526</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>243.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>17/19</td>\n",
       "      <td></td>\n",
       "      <td>rs77999481</td>\n",
       "      <td>tolerated(0.24)</td>\n",
       "      <td>benign(0.003)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946445</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>48.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>rs143216532</td>\n",
       "      <td>tolerated(0.05)</td>\n",
       "      <td>benign(0.243)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>952461</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>34.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>10/19</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.825)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>953810</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>41.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>8/19</td>\n",
       "      <td></td>\n",
       "      <td>COSM4572774</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.854)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>965129</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>51.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>ENSG00000187961</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12/12</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>tolerated(0.26)</td>\n",
       "      <td>probably_damaging(0.931)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM     POS ID REF ALT    QUAL FILTER          FORMAT  SRR12043174  \\\n",
       "35  chr1  945526  .   G   A  243.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "36  chr1  946445  .   C   T   48.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "38  chr1  952461  .   A   G   34.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "39  chr1  953810  .   A   G   41.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "49  chr1  965129  .   G   A   51.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "\n",
       "    SRR12043175  ...  SYMBOL             Gene Feature_type          Feature  \\\n",
       "35  ./.:.:.:.:.  ...   NOC2L  ENSG00000188976   Transcript  ENST00000327044   \n",
       "36  ./.:.:.:.:.  ...   NOC2L  ENSG00000188976   Transcript  ENST00000327044   \n",
       "38  ./.:.:.:.:.  ...   NOC2L  ENSG00000188976   Transcript  ENST00000327044   \n",
       "39  ./.:.:.:.:.  ...   NOC2L  ENSG00000188976   Transcript  ENST00000327044   \n",
       "49  ./.:.:.:.:.  ...  KLHL17  ENSG00000187961   Transcript  ENST00000338591   \n",
       "\n",
       "           BIOTYPE   EXON INTRON Existing_variation             SIFT  \\\n",
       "35  protein_coding  17/19                rs77999481  tolerated(0.24)   \n",
       "36  protein_coding  15/19               rs143216532  tolerated(0.05)   \n",
       "38  protein_coding  10/19                             deleterious(0)   \n",
       "39  protein_coding   8/19               COSM4572774   deleterious(0)   \n",
       "49  protein_coding  12/12                            tolerated(0.26)   \n",
       "\n",
       "                    PolyPhen  \n",
       "35             benign(0.003)  \n",
       "36             benign(0.243)  \n",
       "38  possibly_damaging(0.825)  \n",
       "39  possibly_damaging(0.854)  \n",
       "49  probably_damaging(0.931)  \n",
       "\n",
       "[5 rows x 87 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missense_variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9347"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#to count no of missense variants\n",
    "len(missense_variants.axes[0]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To filter LOF mutations: missense varinats having \"deleterious\" effect in SIFT and \"damaging\" effect in polyphen database\n",
    "LOF_mutations = removed_columns[removed_columns['Consequence'].str.contains(\"missense_variant\") & removed_columns['SIFT'].str.contains(\"deleterious\") & removed_columns['PolyPhen'].str.contains(\"damaging\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SRR12043174</th>\n",
       "      <th>SRR12043175</th>\n",
       "      <th>...</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>952461</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>34.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>10/19</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.825)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>953810</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>41.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>8/19</td>\n",
       "      <td></td>\n",
       "      <td>COSM4572774</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.854)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999877</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>40.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>HES4</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.756)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043289</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>47.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>8/36</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.973)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1047863</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>231.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>22/36</td>\n",
       "      <td></td>\n",
       "      <td>rs142620337</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.997)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS ID REF ALT    QUAL FILTER          FORMAT  SRR12043174  \\\n",
       "38  chr1   952461  .   A   G   34.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "39  chr1   953810  .   A   G   41.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "65  chr1   999877  .   C   T   40.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "78  chr1  1043289  .   T   C   47.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "80  chr1  1047863  .   C   T  231.64   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "\n",
       "    SRR12043175  ... SYMBOL             Gene Feature_type          Feature  \\\n",
       "38  ./.:.:.:.:.  ...  NOC2L  ENSG00000188976   Transcript  ENST00000327044   \n",
       "39  ./.:.:.:.:.  ...  NOC2L  ENSG00000188976   Transcript  ENST00000327044   \n",
       "65  ./.:.:.:.:.  ...   HES4  ENSG00000188290   Transcript  ENST00000304952   \n",
       "78  ./.:.:.:.:.  ...   AGRN  ENSG00000188157   Transcript  ENST00000379370   \n",
       "80  ./.:.:.:.:.  ...   AGRN  ENSG00000188157   Transcript  ENST00000379370   \n",
       "\n",
       "           BIOTYPE   EXON INTRON Existing_variation            SIFT  \\\n",
       "38  protein_coding  10/19                            deleterious(0)   \n",
       "39  protein_coding   8/19               COSM4572774  deleterious(0)   \n",
       "65  protein_coding    1/4                            deleterious(0)   \n",
       "78  protein_coding   8/36                            deleterious(0)   \n",
       "80  protein_coding  22/36               rs142620337  deleterious(0)   \n",
       "\n",
       "                    PolyPhen  \n",
       "38  possibly_damaging(0.825)  \n",
       "39  possibly_damaging(0.854)  \n",
       "65  possibly_damaging(0.756)  \n",
       "78  probably_damaging(0.973)  \n",
       "80  probably_damaging(0.997)  \n",
       "\n",
       "[5 rows x 87 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LOF_mutations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2867"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#To count no of LOF mutations\n",
    "len(LOF_mutations.axes[0]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for rearranging columns of LOF output file \n",
    "def movecol(df, cols_to_move=[], ref_col='', place='After'):\n",
    "    \n",
    "    cols = df.columns.tolist()\n",
    "    if place == 'After':\n",
    "        seg1 = cols[:list(cols).index(ref_col) + 1]\n",
    "        seg2 = cols_to_move\n",
    "    if place == 'Before':\n",
    "        seg1 = cols[:list(cols).index(ref_col)]\n",
    "        seg2 = cols_to_move + [ref_col]\n",
    "    \n",
    "    seg1 = [i for i in seg1 if i not in seg2]\n",
    "    seg3 = [i for i in cols if i not in seg1 + seg2]\n",
    "    \n",
    "    return(df[seg1 + seg2 + seg3])\n",
    "Arranged_LOF_cols = movecol(LOF_mutations, \n",
    "             cols_to_move=[\"Consequence\",\"SYMBOL\",\"Gene\",\"Feature_type\",\"Feature\",\"BIOTYPE\",\"EXON\",\"INTRON\",\"Existing_variation\",\"SIFT\",\"PolyPhen\"], \n",
    "             ref_col='FORMAT',\n",
    "             place='After')\n",
    "#cols_to_move : name of columns to move\n",
    "#ref_col : column where we want to move columns\n",
    "#place : columns to be moved after or Before ref_col"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7721278</th>\n",
       "      <th>SRR7721279</th>\n",
       "      <th>SRR7721280</th>\n",
       "      <th>SRR7721281</th>\n",
       "      <th>SRR7721282</th>\n",
       "      <th>SRR7721283</th>\n",
       "      <th>SRR7721284</th>\n",
       "      <th>SRR7721285</th>\n",
       "      <th>SRR7721286</th>\n",
       "      <th>SRR7721287</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>952461</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>34.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>953810</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>41.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999877</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>40.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HES4</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043289</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>47.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1047863</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>231.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS ID REF ALT    QUAL FILTER          FORMAT       Consequence  \\\n",
       "38  chr1   952461  .   A   G   34.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "39  chr1   953810  .   A   G   41.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "65  chr1   999877  .   C   T   40.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "78  chr1  1043289  .   T   C   47.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "80  chr1  1047863  .   C   T  231.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "\n",
       "   SYMBOL  ...   SRR7721278   SRR7721279   SRR7721280   SRR7721281  \\\n",
       "38  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "39  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "65   HES4  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "78   AGRN  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "80   AGRN  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "     SRR7721282   SRR7721283   SRR7721284   SRR7721285   SRR7721286  \\\n",
       "38  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "39  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "65  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "78  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "80  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "     SRR7721287  \n",
       "38  ./.:.:.:.:.  \n",
       "39  ./.:.:.:.:.  \n",
       "65  ./.:.:.:.:.  \n",
       "78  ./.:.:.:.:.  \n",
       "80  ./.:.:.:.:.  \n",
       "\n",
       "[5 rows x 87 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Arranged_LOF_cols.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "#print output\n",
    "Arranged_LOF_cols.to_csv('Asthma_LOF_mutations.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#to print only missesnse variants\n",
    "#k1 = k[k['Consequence'].str.contains(\"missense_variant\")] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    Consequence  \\\n",
      "38                             missense_variant   \n",
      "39                             missense_variant   \n",
      "65                             missense_variant   \n",
      "366                            missense_variant   \n",
      "405                            missense_variant   \n",
      "410                            missense_variant   \n",
      "832                            missense_variant   \n",
      "984                            missense_variant   \n",
      "1241                           missense_variant   \n",
      "1281     missense_variant&splice_region_variant   \n",
      "1291                           missense_variant   \n",
      "1493                           missense_variant   \n",
      "1502                           missense_variant   \n",
      "1667                           missense_variant   \n",
      "1668                           missense_variant   \n",
      "1838                           missense_variant   \n",
      "1885                           missense_variant   \n",
      "2010                           missense_variant   \n",
      "2239                           missense_variant   \n",
      "2240                           missense_variant   \n",
      "2433                           missense_variant   \n",
      "2605                           missense_variant   \n",
      "2811                           missense_variant   \n",
      "2913                           missense_variant   \n",
      "3324                           missense_variant   \n",
      "3377                           missense_variant   \n",
      "3591                           missense_variant   \n",
      "3637                           missense_variant   \n",
      "3953                           missense_variant   \n",
      "4129                           missense_variant   \n",
      "...                                         ...   \n",
      "128415                         missense_variant   \n",
      "128911                         missense_variant   \n",
      "129068                         missense_variant   \n",
      "129128                         missense_variant   \n",
      "129600                         missense_variant   \n",
      "129687                         missense_variant   \n",
      "129822                         missense_variant   \n",
      "129905  missense_variant&NMD_transcript_variant   \n",
      "130420                         missense_variant   \n",
      "130422                         missense_variant   \n",
      "130606                         missense_variant   \n",
      "130704                         missense_variant   \n",
      "130736                         missense_variant   \n",
      "130970                         missense_variant   \n",
      "131148                         missense_variant   \n",
      "131188                         missense_variant   \n",
      "131382                         missense_variant   \n",
      "131614                         missense_variant   \n",
      "131808   missense_variant&splice_region_variant   \n",
      "131810                         missense_variant   \n",
      "131937                         missense_variant   \n",
      "132049                         missense_variant   \n",
      "132731                         missense_variant   \n",
      "132798                         missense_variant   \n",
      "132914                         missense_variant   \n",
      "133045                         missense_variant   \n",
      "133611                         missense_variant   \n",
      "133671                         missense_variant   \n",
      "134081                         missense_variant   \n",
      "134697                         missense_variant   \n",
      "\n",
      "                                    SIFT                  PolyPhen  \n",
      "38                        deleterious(0)  possibly_damaging(0.825)  \n",
      "39                        deleterious(0)  possibly_damaging(0.854)  \n",
      "65                        deleterious(0)  possibly_damaging(0.756)  \n",
      "366                    deleterious(0.03)   possibly_damaging(0.74)  \n",
      "405                    deleterious(0.01)   possibly_damaging(0.84)  \n",
      "410        deleterious_low_confidence(0)  possibly_damaging(0.646)  \n",
      "832                       deleterious(0)  possibly_damaging(0.878)  \n",
      "984                    deleterious(0.01)  possibly_damaging(0.723)  \n",
      "1241                      deleterious(0)  possibly_damaging(0.725)  \n",
      "1281                   deleterious(0.01)  possibly_damaging(0.686)  \n",
      "1291                   deleterious(0.02)  possibly_damaging(0.893)  \n",
      "1493                   deleterious(0.01)  possibly_damaging(0.737)  \n",
      "1502                      deleterious(0)  possibly_damaging(0.573)  \n",
      "1667                      deleterious(0)  possibly_damaging(0.782)  \n",
      "1668                      deleterious(0)  possibly_damaging(0.695)  \n",
      "1838                   deleterious(0.03)  possibly_damaging(0.706)  \n",
      "1885       deleterious_low_confidence(0)  possibly_damaging(0.881)  \n",
      "2010                   deleterious(0.03)  possibly_damaging(0.596)  \n",
      "2239                      deleterious(0)   possibly_damaging(0.68)  \n",
      "2240                   deleterious(0.04)  possibly_damaging(0.486)  \n",
      "2433                      deleterious(0)  possibly_damaging(0.812)  \n",
      "2605                   deleterious(0.01)  possibly_damaging(0.827)  \n",
      "2811                      deleterious(0)  possibly_damaging(0.888)  \n",
      "2913                   deleterious(0.04)  possibly_damaging(0.902)  \n",
      "3324                   deleterious(0.01)  possibly_damaging(0.487)  \n",
      "3377                      deleterious(0)  possibly_damaging(0.897)  \n",
      "3591                      deleterious(0)  possibly_damaging(0.897)  \n",
      "3637       deleterious_low_confidence(0)  possibly_damaging(0.653)  \n",
      "3953                   deleterious(0.02)  possibly_damaging(0.847)  \n",
      "4129                      deleterious(0)  possibly_damaging(0.888)  \n",
      "...                                  ...                       ...  \n",
      "128415                    deleterious(0)  possibly_damaging(0.813)  \n",
      "128911                 deleterious(0.01)  possibly_damaging(0.474)  \n",
      "129068                 deleterious(0.02)  possibly_damaging(0.518)  \n",
      "129128                 deleterious(0.01)  possibly_damaging(0.899)  \n",
      "129600                    deleterious(0)  possibly_damaging(0.694)  \n",
      "129687                 deleterious(0.03)  possibly_damaging(0.683)  \n",
      "129822                    deleterious(0)  possibly_damaging(0.598)  \n",
      "129905  deleterious_low_confidence(0.01)  possibly_damaging(0.845)  \n",
      "130420                 deleterious(0.05)  possibly_damaging(0.741)  \n",
      "130422                    deleterious(0)  possibly_damaging(0.453)  \n",
      "130606                    deleterious(0)   possibly_damaging(0.88)  \n",
      "130704                 deleterious(0.01)  possibly_damaging(0.629)  \n",
      "130736                 deleterious(0.04)  possibly_damaging(0.649)  \n",
      "130970                    deleterious(0)  possibly_damaging(0.669)  \n",
      "131148                    deleterious(0)  possibly_damaging(0.907)  \n",
      "131188                 deleterious(0.01)  possibly_damaging(0.875)  \n",
      "131382                 deleterious(0.03)  possibly_damaging(0.825)  \n",
      "131614                    deleterious(0)  possibly_damaging(0.893)  \n",
      "131808                 deleterious(0.01)  possibly_damaging(0.896)  \n",
      "131810                 deleterious(0.02)  possibly_damaging(0.544)  \n",
      "131937                 deleterious(0.02)  possibly_damaging(0.468)  \n",
      "132049                 deleterious(0.04)  possibly_damaging(0.543)  \n",
      "132731                 deleterious(0.02)   possibly_damaging(0.89)  \n",
      "132798                    deleterious(0)  possibly_damaging(0.849)  \n",
      "132914                 deleterious(0.03)  possibly_damaging(0.718)  \n",
      "133045                 deleterious(0.02)  possibly_damaging(0.877)  \n",
      "133611                 deleterious(0.05)  possibly_damaging(0.772)  \n",
      "133671                    deleterious(0)  possibly_damaging(0.713)  \n",
      "134081                 deleterious(0.02)  possibly_damaging(0.665)  \n",
      "134697     deleterious_low_confidence(0)  possibly_damaging(0.537)  \n",
      "\n",
      "[930 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "#print specific columns of datafrrame\n",
    "#print(df3[['Consequence','SIFT','PolyPhen']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-30-f4dcd921490f>, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-30-f4dcd921490f>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    df3 = k[k['Consequence'].str.contains(\"missense_variant\") | k['SIFT'].str.contains(\"deleterious(0)\")] | k['Polyphen'].str.contains(\"possibly_damaging(0.825)\")]\u001b[0m\n\u001b[0m                                                                                                                                                                  ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "#to print vaariants by or (|)\n",
    "#df3 = k[k['Consequence'].str.contains(\"missense_variant\") | k['SIFT'].str.contains(\"deleterious(0)\")] | k['Polyphen'].str.contains(\"possibly_damaging(0.825)\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
