{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import numpy as np\n",
    "from pandas import Series"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Open filtered fie from control data from diseasea data\n",
    "df = pd.read_csv('Filtered_only_Asthma_varinats.vcf', sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SRR12043174</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7721278</th>\n",
       "      <th>SRR7721279</th>\n",
       "      <th>SRR7721280</th>\n",
       "      <th>SRR7721281</th>\n",
       "      <th>SRR7721282</th>\n",
       "      <th>SRR7721283</th>\n",
       "      <th>SRR7721284</th>\n",
       "      <th>SRR7721285</th>\n",
       "      <th>SRR7721286</th>\n",
       "      <th>SRR7721287</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135195</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>102.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BaseQRankSum=-0.524;ExcessHet=3.0103;FS=0;MQ=6...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629190</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>194.96</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=3.611...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629241</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>199.96</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=34.95;SOR=2.833...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629277</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>69.32</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ExcessHet=3.0103;FS=0;MQ=60;QD=34.66;SOR=2.303...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS ID REF ALT    QUAL FILTER  \\\n",
       "0  chr1  135163  .   C   T  105.84   PASS   \n",
       "1  chr1  135195  .   A   G  102.64   PASS   \n",
       "2  chr1  629190  .   C   T  194.96   PASS   \n",
       "3  chr1  629241  .   C   T  199.96   PASS   \n",
       "4  chr1  629277  .   C   T   69.32   PASS   \n",
       "\n",
       "                                                INFO          FORMAT  \\\n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  GT:AD:DP:GQ:PL   \n",
       "1  BaseQRankSum=-0.524;ExcessHet=3.0103;FS=0;MQ=6...  GT:AD:DP:GQ:PL   \n",
       "2  ExcessHet=3.0103;FS=0;MQ=60;QD=25.36;SOR=3.611...  GT:AD:DP:GQ:PL   \n",
       "3  ExcessHet=3.0103;FS=0;MQ=60;QD=34.95;SOR=2.833...  GT:AD:DP:GQ:PL   \n",
       "4  ExcessHet=3.0103;FS=0;MQ=60;QD=34.66;SOR=2.303...  GT:AD:DP:GQ:PL   \n",
       "\n",
       "   SRR12043174  ...   SRR7721278   SRR7721279            SRR7721280  \\\n",
       "0  ./.:.:.:.:.  ...  ./.:.:.:.:.  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34   \n",
       "1  ./.:.:.:.:.  ...  ./.:.:.:.:.  ./.:.:.:.:.           ./.:.:.:.:.   \n",
       "2  ./.:.:.:.:.  ...  ./.:.:.:.:.  ./.:.:.:.:.           ./.:.:.:.:.   \n",
       "3  ./.:.:.:.:.  ...  ./.:.:.:.:.  ./.:.:.:.:.           ./.:.:.:.:.   \n",
       "4  ./.:.:.:.:.  ...  ./.:.:.:.:.  ./.:.:.:.:.           ./.:.:.:.:.   \n",
       "\n",
       "            SRR7721281   SRR7721282   SRR7721283   SRR7721284   SRR7721285  \\\n",
       "0  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "1          ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "2          ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "3          ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "4          ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "    SRR7721286   SRR7721287  \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "1  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "2  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "3  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "4  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "\n",
       "[5 rows x 77 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "134779"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# To print no of rows\n",
    "len(df.axes[0]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "# info column have many consequences seperated by comma for one mutation to split (comma) that add to next next row by keeping all data\n",
    "s = df['INFO'].str.split(',').apply(Series, 1).stack() # split by comma\n",
    "s.index = s.index.droplevel(-1) # index splitted column\n",
    "s.name = 'INFO' # rename column as INFO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...\n",
       "0    T|upstream_gene_variant|MODIFIER|RP11-34P13.7|...\n",
       "0    T|downstream_gene_variant|MODIFIER|RP11-34P13....\n",
       "0    T|non_coding_transcript_exon_variant|MODIFIER|...\n",
       "0    T|downstream_gene_variant|MODIFIER|RP11-34P13....\n",
       "Name: INFO, dtype: object"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To delete orignal INFO column from main df file\n",
    "del df['INFO']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "# join spliited info column with main df dataframe\n",
    "df1 = df.join(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SRR12043174</th>\n",
       "      <th>SRR12043175</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7721279</th>\n",
       "      <th>SRR7721280</th>\n",
       "      <th>SRR7721281</th>\n",
       "      <th>SRR7721282</th>\n",
       "      <th>SRR7721283</th>\n",
       "      <th>SRR7721284</th>\n",
       "      <th>SRR7721285</th>\n",
       "      <th>SRR7721286</th>\n",
       "      <th>SRR7721287</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|RP11-34P13.7|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|RP11-34P13....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|RP11-34P13....</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS ID REF ALT    QUAL FILTER          FORMAT  SRR12043174  \\\n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "\n",
       "   SRR12043175  ...   SRR7721279            SRR7721280           SRR7721281  \\\n",
       "0  ./.:.:.:.:.  ...  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0   \n",
       "0  ./.:.:.:.:.  ...  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0   \n",
       "0  ./.:.:.:.:.  ...  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0   \n",
       "0  ./.:.:.:.:.  ...  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0   \n",
       "0  ./.:.:.:.:.  ...  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0   \n",
       "\n",
       "    SRR7721282   SRR7721283   SRR7721284   SRR7721285   SRR7721286  \\\n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "    SRR7721287                                               INFO  \n",
       "0  ./.:.:.:.:.  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  \n",
       "0  ./.:.:.:.:.  T|upstream_gene_variant|MODIFIER|RP11-34P13.7|...  \n",
       "0  ./.:.:.:.:.  T|downstream_gene_variant|MODIFIER|RP11-34P13....  \n",
       "0  ./.:.:.:.:.  T|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "0  ./.:.:.:.:.  T|downstream_gene_variant|MODIFIER|RP11-34P13....  \n",
       "\n",
       "[5 rows x 77 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "987832"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#len(df1.axes[0]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|RP11-34P13.7|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|RP11-34P13....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|RP11-34P13....</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS                                               INFO\n",
       "0  chr1  135163  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...\n",
       "0  chr1  135163  T|upstream_gene_variant|MODIFIER|RP11-34P13.7|...\n",
       "0  chr1  135163  T|downstream_gene_variant|MODIFIER|RP11-34P13....\n",
       "0  chr1  135163  T|non_coding_transcript_exon_variant|MODIFIER|...\n",
       "0  chr1  135163  T|downstream_gene_variant|MODIFIER|RP11-34P13...."
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df1[['CHROM','POS','INFO']].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Again split INFo column by bar(|) to seperate all the colums\n",
    "k = df1.INFO.str.split(\"|\",expand=True,)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>54</th>\n",
       "      <th>55</th>\n",
       "      <th>56</th>\n",
       "      <th>57</th>\n",
       "      <th>58</th>\n",
       "      <th>59</th>\n",
       "      <th>60</th>\n",
       "      <th>61</th>\n",
       "      <th>62</th>\n",
       "      <th>63</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000453576</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-34P13.14</td>\n",
       "      <td>ENSG00000239906</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000493797</td>\n",
       "      <td>antisense</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-34P13.15</td>\n",
       "      <td>ENSG00000268903</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000494149</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-34P13.16</td>\n",
       "      <td>ENSG00000269981</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000595919</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  0   \\\n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...   \n",
       "0                                                  T   \n",
       "0                                                  T   \n",
       "0                                                  T   \n",
       "0                                                  T   \n",
       "\n",
       "                                   1         2              3   \\\n",
       "0             downstream_gene_variant  MODIFIER         CICP27   \n",
       "0               upstream_gene_variant  MODIFIER   RP11-34P13.7   \n",
       "0             downstream_gene_variant  MODIFIER  RP11-34P13.14   \n",
       "0  non_coding_transcript_exon_variant  MODIFIER  RP11-34P13.15   \n",
       "0             downstream_gene_variant  MODIFIER  RP11-34P13.16   \n",
       "\n",
       "                4           5                6                     7    8  9   \\\n",
       "0  ENSG00000233750  Transcript  ENST00000442987  processed_pseudogene           \n",
       "0  ENSG00000238009  Transcript  ENST00000453576               lincRNA           \n",
       "0  ENSG00000239906  Transcript  ENST00000493797             antisense           \n",
       "0  ENSG00000268903  Transcript  ENST00000494149  processed_pseudogene  1/1      \n",
       "0  ENSG00000269981  Transcript  ENST00000595919  processed_pseudogene           \n",
       "\n",
       "   ...      54   55 56 57 58 59 60 61 62 63  \n",
       "0  ...  0.0923  EAS                          \n",
       "0  ...  0.0923  EAS                          \n",
       "0  ...  0.0923  EAS                          \n",
       "0  ...  0.0923  EAS                          \n",
       "0  ...  0.0923  EAS                          \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "64"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#len(k.axes[1]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To add column names to splitted INFO columns\n",
    "k.columns = [\"Allele\",\"Consequence\",\"IMPACT\",\"SYMBOL\",\"Gene\",\"Feature_type\",\"Feature\",\"BIOTYPE\",\"EXON\",\"INTRON\",\"HGVSc\",\"HGVSp\",\"cDNA_position\",\"CDS_position\",\"Protein_position\",\"Amino_acids\",\"Codons\",\"Existing_variation\",\"DISTANCE\",\"STRAND\",\"FLAGS\",\"VARIANT_CLASS\",\"SYMBOL_SOURCE\",\"HGNC_ID\",\"CANONICAL\",\"TSL\",\"APPRIS\",\"CCDS\",\"ENSP\",\"SWISSPROT\",\"TREMBL\",\"UNIPARC\",\"GENE_PHENO\",\"SIFT\",\"PolyPhen\",\"DOMAINS\",\"HGVS_OFFSET\",\"AF\",\"AFR_AF\",\"AMR_AF\",\"EAS_AF\",\"EUR_AF\",\"SAS_AF\",\"AA_AF\",\"EA_AF\",\"ExAC_AF\",\"ExAC_Adj_AF\",\"ExAC_AFR_AF\",\"ExAC_AMR_AF\",\"ExAC_EAS_AF\",\"ExAC_FIN_AF\",\"ExAC_NFE_AF\",\"ExAC_OTH_AF\",\"ExAC_SAS_AF\",\"MAX_AF\",\"MAX_AF_POPS\",\"CLIN_SIG\",\"SOMATIC\",\"PHENO\",\"PUBMED\",\"MOTIF_NAME\",\"MOTIF_POS\",\"HIGH_INF_POS\",\"MOTIF_SCORE_CHANGE\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000453576</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-34P13.14</td>\n",
       "      <td>ENSG00000239906</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000493797</td>\n",
       "      <td>antisense</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-34P13.15</td>\n",
       "      <td>ENSG00000268903</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000494149</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-34P13.16</td>\n",
       "      <td>ENSG00000269981</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000595919</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>0.0923</td>\n",
       "      <td>EAS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Allele  \\\n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...   \n",
       "0                                                  T   \n",
       "0                                                  T   \n",
       "0                                                  T   \n",
       "0                                                  T   \n",
       "\n",
       "                          Consequence    IMPACT         SYMBOL  \\\n",
       "0             downstream_gene_variant  MODIFIER         CICP27   \n",
       "0               upstream_gene_variant  MODIFIER   RP11-34P13.7   \n",
       "0             downstream_gene_variant  MODIFIER  RP11-34P13.14   \n",
       "0  non_coding_transcript_exon_variant  MODIFIER  RP11-34P13.15   \n",
       "0             downstream_gene_variant  MODIFIER  RP11-34P13.16   \n",
       "\n",
       "              Gene Feature_type          Feature               BIOTYPE EXON  \\\n",
       "0  ENSG00000233750   Transcript  ENST00000442987  processed_pseudogene        \n",
       "0  ENSG00000238009   Transcript  ENST00000453576               lincRNA        \n",
       "0  ENSG00000239906   Transcript  ENST00000493797             antisense        \n",
       "0  ENSG00000268903   Transcript  ENST00000494149  processed_pseudogene  1/1   \n",
       "0  ENSG00000269981   Transcript  ENST00000595919  processed_pseudogene        \n",
       "\n",
       "  INTRON  ...  MAX_AF MAX_AF_POPS CLIN_SIG SOMATIC PHENO PUBMED MOTIF_NAME  \\\n",
       "0         ...  0.0923         EAS                                            \n",
       "0         ...  0.0923         EAS                                            \n",
       "0         ...  0.0923         EAS                                            \n",
       "0         ...  0.0923         EAS                                            \n",
       "0         ...  0.0923         EAS                                            \n",
       "\n",
       "  MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE  \n",
       "0                                            \n",
       "0                                            \n",
       "0                                            \n",
       "0                                            \n",
       "0                                            \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To filter only important columns from splitted INFO column and saved in p variable\n",
    "p = k[[\"Consequence\",\"SYMBOL\",\"Gene\",\"Feature_type\",\"Feature\",\"BIOTYPE\",\"EXON\",\"INTRON\",\"Existing_variation\",\"SIFT\",\"PolyPhen\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000453576</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>RP11-34P13.14</td>\n",
       "      <td>ENSG00000239906</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000493797</td>\n",
       "      <td>antisense</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>RP11-34P13.15</td>\n",
       "      <td>ENSG00000268903</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000494149</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>RP11-34P13.16</td>\n",
       "      <td>ENSG00000269981</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000595919</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Consequence         SYMBOL             Gene  \\\n",
       "0             downstream_gene_variant         CICP27  ENSG00000233750   \n",
       "0               upstream_gene_variant   RP11-34P13.7  ENSG00000238009   \n",
       "0             downstream_gene_variant  RP11-34P13.14  ENSG00000239906   \n",
       "0  non_coding_transcript_exon_variant  RP11-34P13.15  ENSG00000268903   \n",
       "0             downstream_gene_variant  RP11-34P13.16  ENSG00000269981   \n",
       "\n",
       "  Feature_type          Feature               BIOTYPE EXON INTRON  \\\n",
       "0   Transcript  ENST00000442987  processed_pseudogene               \n",
       "0   Transcript  ENST00000453576               lincRNA               \n",
       "0   Transcript  ENST00000493797             antisense               \n",
       "0   Transcript  ENST00000494149  processed_pseudogene  1/1          \n",
       "0   Transcript  ENST00000595919  processed_pseudogene               \n",
       "\n",
       "  Existing_variation SIFT PolyPhen  \n",
       "0        rs374499151                \n",
       "0        rs374499151                \n",
       "0        rs374499151                \n",
       "0        rs374499151                \n",
       "0        rs374499151                "
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "#join splitted INFO column by bar (|) with main vcf file \n",
    "final_INFO_splitted_file = df1.join(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SRR12043174</th>\n",
       "      <th>SRR12043175</th>\n",
       "      <th>...</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000442987</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000453576</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>RP11-34P13.14</td>\n",
       "      <td>ENSG00000239906</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000493797</td>\n",
       "      <td>antisense</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>RP11-34P13.15</td>\n",
       "      <td>ENSG00000268903</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000494149</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>...</td>\n",
       "      <td>RP11-34P13.16</td>\n",
       "      <td>ENSG00000269981</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000595919</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs374499151</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 88 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS ID REF ALT    QUAL FILTER          FORMAT  SRR12043174  \\\n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL  ./.:.:.:.:.   \n",
       "\n",
       "   SRR12043175  ...         SYMBOL             Gene Feature_type  \\\n",
       "0  ./.:.:.:.:.  ...         CICP27  ENSG00000233750   Transcript   \n",
       "0  ./.:.:.:.:.  ...   RP11-34P13.7  ENSG00000238009   Transcript   \n",
       "0  ./.:.:.:.:.  ...  RP11-34P13.14  ENSG00000239906   Transcript   \n",
       "0  ./.:.:.:.:.  ...  RP11-34P13.15  ENSG00000268903   Transcript   \n",
       "0  ./.:.:.:.:.  ...  RP11-34P13.16  ENSG00000269981   Transcript   \n",
       "\n",
       "           Feature               BIOTYPE EXON INTRON Existing_variation SIFT  \\\n",
       "0  ENST00000442987  processed_pseudogene                    rs374499151        \n",
       "0  ENST00000453576               lincRNA                    rs374499151        \n",
       "0  ENST00000493797             antisense                    rs374499151        \n",
       "0  ENST00000494149  processed_pseudogene  1/1               rs374499151        \n",
       "0  ENST00000595919  processed_pseudogene                    rs374499151        \n",
       "\n",
       "  PolyPhen  \n",
       "0           \n",
       "0           \n",
       "0           \n",
       "0           \n",
       "0           \n",
       "\n",
       "[5 rows x 88 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_INFO_splitted_file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...\n",
       "0    BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...\n",
       "0    BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...\n",
       "0    BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...\n",
       "0    BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...\n",
       "Name: INFO, dtype: object"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##final_INFO_splitted_file[\"INFO\"].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'final_INFO_splitted_file' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-2-060ede42a257>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m     \u001b[0;32mreturn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mseg1\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mseg2\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mseg3\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 16\u001b[0;31m Arranged_cols = movecol(final_INFO_splitted_file, \n\u001b[0m\u001b[1;32m     17\u001b[0m              \u001b[0mcols_to_move\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Consequence\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"SYMBOL\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"Gene\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"Feature_type\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"Feature\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"BIOTYPE\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"EXON\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"INTRON\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"Existing_variation\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"SIFT\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"PolyPhen\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m              \u001b[0mref_col\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'FORMAT'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'final_INFO_splitted_file' is not defined"
     ]
    }
   ],
   "source": [
    "# for rearranging columns of LOF output file \n",
    "def movecol(df, cols_to_move=[], ref_col='', place='After'):\n",
    "    \n",
    "    cols = df.columns.tolist()\n",
    "    if place == 'After':\n",
    "        seg1 = cols[:list(cols).index(ref_col) + 1]\n",
    "        seg2 = cols_to_move\n",
    "    if place == 'Before':\n",
    "        seg1 = cols[:list(cols).index(ref_col)]\n",
    "        seg2 = cols_to_move + [ref_col]\n",
    "    \n",
    "    seg1 = [i for i in seg1 if i not in seg2]\n",
    "    seg3 = [i for i in cols if i not in seg1 + seg2]\n",
    "    \n",
    "    return(df[seg1 + seg2 + seg3])\n",
    "Arranged_cols = movecol(final_INFO_splitted_file, \n",
    "             cols_to_move=[\"Consequence\",\"SYMBOL\",\"Gene\",\"Feature_type\",\"Feature\",\"BIOTYPE\",\"EXON\",\"INTRON\",\"Existing_variation\",\"SIFT\",\"PolyPhen\"], \n",
    "             ref_col='FORMAT',\n",
    "             place='After')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7721279</th>\n",
       "      <th>SRR7721280</th>\n",
       "      <th>SRR7721281</th>\n",
       "      <th>SRR7721282</th>\n",
       "      <th>SRR7721283</th>\n",
       "      <th>SRR7721284</th>\n",
       "      <th>SRR7721285</th>\n",
       "      <th>SRR7721286</th>\n",
       "      <th>SRR7721287</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>RP11-34P13.14</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>RP11-34P13.15</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>RP11-34P13.16</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 88 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS ID REF ALT    QUAL FILTER          FORMAT  \\\n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "\n",
       "                          Consequence         SYMBOL  ...   SRR7721279  \\\n",
       "0             downstream_gene_variant         CICP27  ...  ./.:.:.:.:.   \n",
       "0               upstream_gene_variant   RP11-34P13.7  ...  ./.:.:.:.:.   \n",
       "0             downstream_gene_variant  RP11-34P13.14  ...  ./.:.:.:.:.   \n",
       "0  non_coding_transcript_exon_variant  RP11-34P13.15  ...  ./.:.:.:.:.   \n",
       "0             downstream_gene_variant  RP11-34P13.16  ...  ./.:.:.:.:.   \n",
       "\n",
       "             SRR7721280           SRR7721281   SRR7721282   SRR7721283  \\\n",
       "0  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "    SRR7721284   SRR7721285   SRR7721286   SRR7721287  \\\n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "                                                INFO  \n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  \n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  \n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  \n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  \n",
       "0  BaseQRankSum=0;ExcessHet=3.0103;FS=0;MQ=60;MQR...  \n",
       "\n",
       "[5 rows x 88 columns]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Arranged_cols.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'Arranged_cols' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-02584261ce0d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# To remove any extra or unwanted columns from Arranged_cols file\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mremoved_columns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mArranged_cols\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdrop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'INFO'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'Arranged_cols' is not defined"
     ]
    }
   ],
   "source": [
    "# To remove any extra or unwanted columns from Arranged_cols file\n",
    "removed_columns = Arranged_cols.drop(['INFO'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7721278</th>\n",
       "      <th>SRR7721279</th>\n",
       "      <th>SRR7721280</th>\n",
       "      <th>SRR7721281</th>\n",
       "      <th>SRR7721282</th>\n",
       "      <th>SRR7721283</th>\n",
       "      <th>SRR7721284</th>\n",
       "      <th>SRR7721285</th>\n",
       "      <th>SRR7721286</th>\n",
       "      <th>SRR7721287</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>CICP27</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>RP11-34P13.14</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>RP11-34P13.15</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135163</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>105.84</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>RP11-34P13.16</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:1,2:3:34:77,0,34</td>\n",
       "      <td>1/1:0,3:3:9:119,9,0</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS ID REF ALT    QUAL FILTER          FORMAT  \\\n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "0  chr1  135163  .   C   T  105.84   PASS  GT:AD:DP:GQ:PL   \n",
       "\n",
       "                          Consequence         SYMBOL  ...   SRR7721278  \\\n",
       "0             downstream_gene_variant         CICP27  ...  ./.:.:.:.:.   \n",
       "0               upstream_gene_variant   RP11-34P13.7  ...  ./.:.:.:.:.   \n",
       "0             downstream_gene_variant  RP11-34P13.14  ...  ./.:.:.:.:.   \n",
       "0  non_coding_transcript_exon_variant  RP11-34P13.15  ...  ./.:.:.:.:.   \n",
       "0             downstream_gene_variant  RP11-34P13.16  ...  ./.:.:.:.:.   \n",
       "\n",
       "    SRR7721279            SRR7721280           SRR7721281   SRR7721282  \\\n",
       "0  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.   \n",
       "0  ./.:.:.:.:.  0/1:1,2:3:34:77,0,34  1/1:0,3:3:9:119,9,0  ./.:.:.:.:.   \n",
       "\n",
       "    SRR7721283   SRR7721284   SRR7721285   SRR7721286   SRR7721287  \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "0  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "\n",
       "[5 rows x 87 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "removed_columns.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "#print output file with all mutations and its consequences\n",
    "removed_columns.to_csv('Asthma_final_mutations.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### LOF analysis satrted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/home/komal.bhosle/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (15,16,17,18,19) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "#LOF analysis input\n",
    "LOF_input = pd.read_csv('Asthma_final_mutations.csv',sep = \",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove unwanted Unnamed: 0 column\n",
    "removed_columns_1 = LOF_input.drop(['Unnamed: 0'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# to find missense mutation\n",
    "missense_variants = removed_columns_1.loc[removed_columns_1['Consequence'].str.contains('missense_variant') == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7721278</th>\n",
       "      <th>SRR7721279</th>\n",
       "      <th>SRR7721280</th>\n",
       "      <th>SRR7721281</th>\n",
       "      <th>SRR7721282</th>\n",
       "      <th>SRR7721283</th>\n",
       "      <th>SRR7721284</th>\n",
       "      <th>SRR7721285</th>\n",
       "      <th>SRR7721286</th>\n",
       "      <th>SRR7721287</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5884</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945526</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>243.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:9,7:16:99:251,0,329</td>\n",
       "      <td>0/1:9,5:14:99:170,0,327</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5903</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945526</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>243.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:9,7:16:99:251,0,329</td>\n",
       "      <td>0/1:9,5:14:99:170,0,327</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5922</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945526</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>243.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:9,7:16:99:251,0,329</td>\n",
       "      <td>0/1:9,5:14:99:170,0,327</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5941</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945526</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>243.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:9,7:16:99:251,0,329</td>\n",
       "      <td>0/1:9,5:14:99:170,0,327</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5960</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945526</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>243.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>0/1:9,7:16:99:251,0,329</td>\n",
       "      <td>0/1:9,5:14:99:170,0,327</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM     POS ID REF ALT    QUAL FILTER          FORMAT  \\\n",
       "5884  chr1  945526  .   G   A  243.64   PASS  GT:AD:DP:GQ:PL   \n",
       "5903  chr1  945526  .   G   A  243.64   PASS  GT:AD:DP:GQ:PL   \n",
       "5922  chr1  945526  .   G   A  243.64   PASS  GT:AD:DP:GQ:PL   \n",
       "5941  chr1  945526  .   G   A  243.64   PASS  GT:AD:DP:GQ:PL   \n",
       "5960  chr1  945526  .   G   A  243.64   PASS  GT:AD:DP:GQ:PL   \n",
       "\n",
       "           Consequence SYMBOL  ...   SRR7721278   SRR7721279  \\\n",
       "5884  missense_variant  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "5903  missense_variant  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "5922  missense_variant  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "5941  missense_variant  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "5960  missense_variant  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "                   SRR7721280               SRR7721281   SRR7721282  \\\n",
       "5884  0/1:9,7:16:99:251,0,329  0/1:9,5:14:99:170,0,327  ./.:.:.:.:.   \n",
       "5903  0/1:9,7:16:99:251,0,329  0/1:9,5:14:99:170,0,327  ./.:.:.:.:.   \n",
       "5922  0/1:9,7:16:99:251,0,329  0/1:9,5:14:99:170,0,327  ./.:.:.:.:.   \n",
       "5941  0/1:9,7:16:99:251,0,329  0/1:9,5:14:99:170,0,327  ./.:.:.:.:.   \n",
       "5960  0/1:9,7:16:99:251,0,329  0/1:9,5:14:99:170,0,327  ./.:.:.:.:.   \n",
       "\n",
       "       SRR7721283   SRR7721284   SRR7721285   SRR7721286   SRR7721287  \n",
       "5884  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "5903  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "5922  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "5941  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "5960  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "\n",
       "[5 rows x 87 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missense_variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "385543"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#To find no of missense variants\n",
    "len(missense_variants.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To find LOF mutations\n",
    "LOF_mutations = removed_columns_1.loc[removed_columns_1['Consequence'].str.contains(\"missense_variant\") & removed_columns_1['SIFT'].str.contains(\"deleterious\") & removed_columns_1['PolyPhen'].str.contains(\"damaging\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7721278</th>\n",
       "      <th>SRR7721279</th>\n",
       "      <th>SRR7721280</th>\n",
       "      <th>SRR7721281</th>\n",
       "      <th>SRR7721282</th>\n",
       "      <th>SRR7721283</th>\n",
       "      <th>SRR7721284</th>\n",
       "      <th>SRR7721285</th>\n",
       "      <th>SRR7721286</th>\n",
       "      <th>SRR7721287</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6831</th>\n",
       "      <td>chr1</td>\n",
       "      <td>952461</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>34.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6834</th>\n",
       "      <td>chr1</td>\n",
       "      <td>952461</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>34.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6837</th>\n",
       "      <td>chr1</td>\n",
       "      <td>952461</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>34.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6840</th>\n",
       "      <td>chr1</td>\n",
       "      <td>953810</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>41.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>chr1</td>\n",
       "      <td>953810</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>41.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM     POS ID REF ALT   QUAL FILTER          FORMAT       Consequence  \\\n",
       "6831  chr1  952461  .   A   G  34.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "6834  chr1  952461  .   A   G  34.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "6837  chr1  952461  .   A   G  34.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "6840  chr1  953810  .   A   G  41.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "6844  chr1  953810  .   A   G  41.64   PASS  GT:AD:DP:GQ:PL  missense_variant   \n",
       "\n",
       "     SYMBOL  ...   SRR7721278   SRR7721279   SRR7721280   SRR7721281  \\\n",
       "6831  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6834  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6837  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6840  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6844  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "       SRR7721282   SRR7721283   SRR7721284   SRR7721285   SRR7721286  \\\n",
       "6831  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6834  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6837  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6840  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6844  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "       SRR7721287  \n",
       "6831  ./.:.:.:.:.  \n",
       "6834  ./.:.:.:.:.  \n",
       "6837  ./.:.:.:.:.  \n",
       "6840  ./.:.:.:.:.  \n",
       "6844  ./.:.:.:.:.  \n",
       "\n",
       "[5 rows x 87 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LOF_mutations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "123173"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# To count no of LOF mutation\n",
    "len(LOF_mutations.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "#LOF_mutations.head(100).to_csv('exp_LOF_head.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "#LOF unique mutations (Remove duplicates)\n",
    "LOF_unique = LOF_mutations.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7721278</th>\n",
       "      <th>SRR7721279</th>\n",
       "      <th>SRR7721280</th>\n",
       "      <th>SRR7721281</th>\n",
       "      <th>SRR7721282</th>\n",
       "      <th>SRR7721283</th>\n",
       "      <th>SRR7721284</th>\n",
       "      <th>SRR7721285</th>\n",
       "      <th>SRR7721286</th>\n",
       "      <th>SRR7721287</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6831</th>\n",
       "      <td>chr1</td>\n",
       "      <td>952461</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>34.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6840</th>\n",
       "      <td>chr1</td>\n",
       "      <td>953810</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>41.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8647</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999877</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>40.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HES4</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8648</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999877</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>40.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HES4</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9067</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043289</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>47.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>...</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "      <td>./.:.:.:.:.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM      POS ID REF ALT   QUAL FILTER          FORMAT  \\\n",
       "6831  chr1   952461  .   A   G  34.64   PASS  GT:AD:DP:GQ:PL   \n",
       "6840  chr1   953810  .   A   G  41.64   PASS  GT:AD:DP:GQ:PL   \n",
       "8647  chr1   999877  .   C   T  40.64   PASS  GT:AD:DP:GQ:PL   \n",
       "8648  chr1   999877  .   C   T  40.64   PASS  GT:AD:DP:GQ:PL   \n",
       "9067  chr1  1043289  .   T   C  47.64   PASS  GT:AD:DP:GQ:PL   \n",
       "\n",
       "           Consequence SYMBOL  ...   SRR7721278   SRR7721279   SRR7721280  \\\n",
       "6831  missense_variant  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6840  missense_variant  NOC2L  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "8647  missense_variant   HES4  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "8648  missense_variant   HES4  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "9067  missense_variant   AGRN  ...  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "       SRR7721281   SRR7721282   SRR7721283   SRR7721284   SRR7721285  \\\n",
       "6831  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "6840  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "8647  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "8648  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "9067  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.  ./.:.:.:.:.   \n",
       "\n",
       "       SRR7721286   SRR7721287  \n",
       "6831  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "6840  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "8647  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "8648  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "9067  ./.:.:.:.:.  ./.:.:.:.:.  \n",
       "\n",
       "[5 rows x 87 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LOF_unique.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11241"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# No of unique LOF mutation\n",
    "len(LOF_unique.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "# output of LOF mutation\n",
    "LOF_unique.head(100).to_csv('unique_exp_LOF_head.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
